---
title: "IAMC- Manuscript1"
author: "Kelsey N. Thompson, Ph.D."
date: "March 19, 2019"
header-includes:
- \usepackage{titling}
- \pretitle{\begin{center}\LARGE\includegraphics[width=12cm]{HarvardChan_logo_center_subbrand_RGB_large.png}\\[\bigskipamount]}
- \posttitle{\end{center}}
output: 
  pdf_document:
    toc: true
    keep_tex: true
    dev: png
---


\newpage
```{r setup_functions, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dpi=600)
library(ggplot2)
library(vegan)
library(ggthemes)
library(ComplexHeatmap)
library(circlize)
library(gridExtra)
library(grid)
library(pheatmap)
library(RColorBrewer)
library(reshape2)
library(scales)
library(gplots)
library(gtools)
library(plyr)
library(dplyr)
library(mgsub)
library(readr)
library(viridis)
library(pairwiseAdonis)
library(kableExtra)
library(randomForest)
library(caret)
library(ROCR)
library(ggridges)

###### functions #######
keep_species <-function(dat_taxa){
  temp = dat_taxa[grepl("s__",rownames(dat_taxa)) & (!grepl("t__",rownames(dat_taxa))) & (!grepl("unclassified$",rownames(dat_taxa))),]
  rownames(temp) = gsub(".*g__.*\\|","",rownames(temp))
  return(temp)
}

is_common <- function(otu_df,cutoff=0.1){
  num_row = dim(otu_df)[1]
  num_col = dim(otu_df)[2]
  otu_barcode = otu_df > 0.0001
  common_index = apply(otu_barcode,1,mean) > cutoff
  return(common_index)
}

top_abun <- function(df,num=10){
  # select top 25 abundunt species
  index = sort(apply(df,1,mean),decreasing=TRUE,index.return=TRUE)$ix[1:num]
  return(1:dim(df)[1] %in% index)
}

top_abun_median <- function(df,num=25){
  # select top 25 abundunt species
  index = sort(apply(df,1,median),decreasing=TRUE,index.return=TRUE)$ix[1:num]
  return(1:dim(df)[1] %in% index)
}


keep_pathways <-function(dat_path){
  temp = dat_path[!grepl("\\|",rownames(dat_path)),]
  # rownames(temp) = gsub(":.*","",rownames(temp))
  return(temp)
}

select_pathways <-function(dat_path,dat_taxa){
  taxa.names = rownames(dat_taxa)
  pathway.names = rownames(dat_path)
  index = rep(FALSE,length(pathway.names))
  for(taxa in taxa.names){
    index = index | grepl(taxa,pathway.names)
  }
  pathway.names.select = pathway.names[index]
  pathway.names.select = unique(gsub("\\|.*","",pathway.names.select))
  dat_path = dat_path[pathway.names.select,]
  # rownames(dat_path) = gsub(":.*","",rownames(dat_path))
  return(dat_path)
}

median_matrix <-function(dat_path,dat_taxa){
  taxa.names = rownames(dat_taxa)
  pathway.names = rownames(dat_path)
  #print(dim(dat_path))
  #print( sum(apply(dat_path>0,1,sum) <= 0 ) )
  
  index = rep(FALSE,length(pathway.names))
  for(taxa in taxa.names){
    index = index | grepl(taxa,pathway.names)
  }
  pathway.names = pathway.names[index]
  dat_path = dat_path[pathway.names,]
  #print(dim(dat_path))
  #print( sum(apply(dat_path>0,1,sum) <= 0 ) )
  
  v_median_all = apply(dat_path,1,median)
  pathway.names.id = gsub("\\|.*","",rownames(dat_path))
  pathway.names.id.unique = unique(pathway.names.id)
  
  out = matrix(0,nrow=length(taxa.names),ncol=length(pathway.names.id.unique))
  
  rownames(out) = taxa.names
  colnames(out) = pathway.names.id.unique
  
  for(taxa in taxa.names){
    for(pathway in pathway.names.id.unique){
      m = v_median_all[grepl(taxa,pathway.names) & pathway == pathway.names.id]
      if(length(m) ==1){
        out[taxa,pathway] = m
      }
      if(length(m) >1){
        print(length(m))
      }
    }
  }
  #out = out[,apply(out,2,median)>0]
  out = out[,names(sort(apply(out,2,sum),decreasing = TRUE)[1:50])]
  return(out)
}

same_median_matrix <- function(dat_median,dat_path){
  taxa.names = rownames(dat_median)
  pathway.names.id.unique = colnames(dat_median)
  
  out = matrix(0,nrow=length(taxa.names),ncol=length(pathway.names.id.unique))
  
  rownames(out) = taxa.names
  colnames(out) = pathway.names.id.unique
  
  v_median_all = apply(dat_path,1,median)
  
  pathway.names = rownames(dat_path)
  pathway.names.id = gsub("\\|.*","",rownames(dat_path))
  
  for(taxa in taxa.names){
    for(pathway in pathway.names.id.unique){
      m = v_median_all[grepl(taxa,pathway.names) & pathway == pathway.names.id]
      if(length(m) ==1){
        out[taxa,pathway] = m
      }
      if(length(m) >1){
        print(length(m))
      }
    }
  }
  return(out)
}

#Set colors to be consistent across comparisons
col_drug = c("DMARD conventional synthetic" = "#FFD700", "DMARD biologic" = "#4B0082", "DMARD targeted synthetic" = "#8A2BE2", "Steroid" = "#FF8C00",  "NSAID" = "#F08080", "other" = "#FF4500", "Not recorded" = "#708090", "PPI" = "#87CEFA", "none" = "#90EE90")
col_diagnosis = c("HC" = "#000080", "RA" = 	"#800080", "AS" = "#DA70D6", "PsA" = "#008000", "NIJP" = "#66CDAA")
col_inflammation = c("Inflammation" = "#B22222", "Some inflammation" = "#FF8C00", "Not inflamed" = "#4682B4", "Not recorded" = "#808080")
col_rb = c("Normal" = "#5F9EA0", "Anemia" = "#FF6347", "Not recorded" = "#696969")


setwd("/Users/kelseythompson/Dropbox (Huttenhower Lab)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional")
```

\newpage
```{r setup_tax_meta, include=FALSE}
map = read.delim("~/Dropbox (Huttenhower Lab)/hutlab/Kelsey/IAMC/metadata/mapping_master.tsv", sep = "\t")
map = map[, -c(1:3)]
map$batch = as.factor(map$batch)

meta_bhm_ncl = read.csv("~/Dropbox (Huttenhower Lab)/hutlab/Kelsey/IAMC/metadata/METADATABHMNCL_DATA_2021-02-17_1322.csv", stringsAsFactors = FALSE)
#Pull out each sample once
bhm_info = subset(meta_bhm_ncl, redcap_repeat_instrument == "")
bhm_info = bhm_info[,colSums(is.na(bhm_info))<nrow(bhm_info)]
#Pull out just the metadata with medication
bhm_med = subset(meta_bhm_ncl, redcap_repeat_instrument == "treatment_and_concomitant_medication_at_time_of_as")
bhm_med = bhm_med[,colSums(is.na(bhm_med))<nrow(bhm_med)]
#Find out which columns are shared between the two datasets
repeat_measure = intersect(colnames(bhm_info), colnames(bhm_med))
repeat_measure = repeat_measure[-c(1)]
bhm_med = bhm_med[, -which(names(bhm_med) %in% repeat_measure)]
bhm = merge(bhm_info, bhm_med, by = "iamc_id", all = T)
bhm$Sample = paste(bhm$iamc_id, bhm$redcap_repeat_instance, sep = "v")
bhm$Sample = gsub("v1", "", bhm$Sample)
bhm$Sample = gsub("vNA", "", bhm$Sample)
row.names(bhm) = bhm$Sample
#Add the cohort within the IAMC to the samples
bhm$Cohort = gsub("\\d", "", row.names(bhm))
bhm$batch = map[ match( bhm$Sample, map$sample_id ), "batch"  ]
#Work on getting the drug category to make sense
bhm$drug_category[is.na(bhm$drug_category)] = "Not recorded"
bhm$drug_category = as.character(bhm$drug_category)
bhm$drug = recode(bhm$drug_category, "1" = "DMARD conventional synthetic", "2" = "DMARD biologic", "3" = "DMARD targeted synthetic", "4" = "Steroid", "5" = "NSAID", "6" = "other", "7" = "none", "8" = "PPI", "9" = "H2", "Not recorded" = "Not recorded", .default = "Not recorded")

#Work on teh daignosis provided- Karim check these: said they were correct
bhm$diagnosis_at_time_of_asses[is.na(bhm$diagnosis_at_time_of_asses)] = "Unknown"
bhm$diagnosis_at_time_of_asses[(bhm$diagnosis_at_time_of_asses== "")] = "Unknown"
bhm$diagnosis_at_time_of_asses = mgsub(bhm$diagnosis_at_time_of_asses, c("  $", " $", "^ "), c("", "", ""))
bhm$diagnosis = recode(bhm$diagnosis_at_time_of_asses, "ACPA+ arthralgia" = "Clinically Suspect Arthralgia", "Ankylosing Spondylitis" = "Ankylosing Spondylitis", "Clinically Suspect Arthralgia" = "Clinically Suspect Arthralgia", "Crystal Arthritis" = "Crystal Arthritis", "Drug induced Inflammatory Arthritis" = "Drug Induced Inflammatory Arthritis", "Drug Induced Inflammatory Arthritis" = "Drug Induced Inflammatory Arthritis", "Fibromyalgia" = "Fibromyalgia", "Gout" = "Gout", "Healthy Control" = "Healthy Control", "Healthy control" = "Healthy Control", "Healthy control.  Blood relative (Daughter) of SWEAC 0213 (BHM00024)" = "Healthy Control", "Healthy control.  Partner of SWEAC 00321 (BHM00205)" = "Healthy Control", "Lupus/Other CTD-Associated" = "Lupus/Other CTD-Associated", "Non-Inflammatory" = "Non-Inflammatory", "Non inflammatory" = "Non-Inflammatory", "PsA" = "PsA", "PSA" = "PsA", "Pseudogout" = "Pseudogout", "Pseudo Gout" = "Pseudogout", "Psoriatic Arthritis" = "PsA", "Ra (1987 & 2010)" = "RA (1987 & 2010)", "RA (1987 & 2010)" = "RA (1987 & 2010)", "RA (1987)" = "RA (1987)", "RA (2010)" = "RA (2010)", "Reactive Arthritis" = "Reactive Arthritis", "Rheumatoid Arthritis" = "RA", "RS3PE" = "RS3PE", "Unclassified Inflammatory Arthritis" = "Unclassified Inflammatory Arthritis", "Undifferentiated Spondylo-Arthropathy" = "Undifferentiated Spondylo-Arthropathy", "Unknown" = "Unknown", "Osteoarthritis" = "Osteoarthritis", "Other Inflammatory Arthritis" = "Other Inflammatory Arthritis", "Inflammatory Arthritis related to Hereditary Haemochromatosis" = "Other Inflammatory Arthritis", "Palindromic Rheumatism" = "Other Inflammatory Arthritis", "Other Inflammatory" = "Other Inflammatory Arthritis", "Undifferentiated Inflammatory Arthritis" = "Clinically Suspect Arthralgia", "Enteropathic Arthritis" = "Enteropathic Arthritis", .default = "Unknown")


bhm$arthritis_type = recode(bhm$diagnosis, "RA (1987 & 2010)" = "RA", "ACPA+ arthralgia" = "Clinically Suspect Arthralgia", "Ankylosing Spondylitis" = "AS", "Clinically Suspect Arthralgia" = "Clinically Suspect Arthralgia", "Crystal Arthritis" = "Crystal Arthritis", "Drug Induced Inflammatory Arthritis" = "other", "Fibromyalgia" = "Non-Inflammatory Joint Pain", "Gout" = "Crystal Arthritis", "Healthy Control" = "Healthy Control", "Lupus/Other CTD-Associated" = "other", "Non-Inflammatory" = "Non-Inflammatory Joint Pain", "PsA" = "PsA", "Pseudogout" = "Crystal Arthritis", "RA (1987 & 2010)" = "RA", "RA (1987)" = "RA", "RA (2010)" = "RA", "Reactive Arthritis" = "other", "RA" = "RA", "RS3PE" = "other", "Unclassified Inflammatory Arthritis" = "Unclassified Inflammatory Arthritis", "Undifferentiated Spondylo-Arthropathy" = "other", "Unknown" = "Unknown", "Osteoarthritis" = "Non-Inflammatory Joint Pain", "Other Inflammatory Arthritis" = "other", "Enteropathic Arthritis" = "other", .default = "Unknown")

library(lubridate)
bhm$DOB = parse_date_time(bhm$date_of_birth_month_year, orders = c("ymd", "dmy", "mdy"))
bhm$DOB = gsub(" UTC", "", bhm$DOB)
bhm$date = parse_date_time(bhm$assessment_date, orders = c("ymd", "dmy", "mdy"))
bhm$date = gsub(" UTC", "", bhm$date)
bhm$age = round(time_length(difftime(as.Date(bhm$date), as.Date(bhm$DOB)), "years"))

#bhm$ccp = paste(bhm$anti_ccp, bhm$arthritis_type, sep = "_")
#bhm$ccp = gsub("Not recorded/ Not known_Healthy Control", "Negative", bhm$ccp)
#bhm$ccp = gsub("_.*", "", bhm$ccp)

meta_oas = read.csv("~/Dropbox (Huttenhower Lab)/hutlab/Kelsey/IAMC/metadata/METADATAOAS_DATA_2021-02-17_1321.csv", head = T, sep = ",")
oas_info = subset(meta_oas, redcap_repeat_instrument == "")
oas_info = oas_info[,colSums(is.na(oas_info))<nrow(oas_info)]
oas_med = subset(meta_oas, redcap_repeat_instrument == "treatment_and_concomitant_medication_at_time_of_as")
oas_med = oas_med[,colSums(is.na(oas_med))<nrow(oas_med)]
repeat_measure = intersect(colnames(oas_info), colnames(oas_med))
repeat_measure = repeat_measure[-c(1)]
oas_med = oas_med[, -which(names(oas_med) %in% repeat_measure)]
oas = merge(oas_info, oas_med, by = "iamc_id", all = TRUE)
oas$Sample = paste(oas$iamc_id, oas$redcap_repeat_instance, sep = "v")
oas$Sample = gsub("v1", "", oas$Sample)
oas$Sample = gsub("vNA", "", oas$Sample)
row.names(oas) = oas$Sample
oas$Cohort = rep("OAS", nrow(oas))
oas$diagnosis = oas$patient_id
oas$diagnosis = gsub("\\d", "", oas$diagnosis)
oas$diagnosis = gsub("HCB", "HC", oas$diagnosis)
oas$drug_category = (ifelse(is.na(as.character(oas$drug_category)), "Not recorded", as.character(oas$drug_category)))
oas$drug = recode(oas$drug_category, "1" = "DMARD conventional synthetic", "2" = "DMARD biologic", "3" = "DMARD targeted synthetic", "4" = "Steroid", "5" = "NSAID", "6" = "other", "7" = "none", "8" = "PPI", "9" = "H2", "Not recorded/Not known" = "Not recorded/Not known", .default = "Not recorded/Not known")
oas$batch = map[ match( oas$Sample, map$sample_id ), "batch"  ]
library(lubridate)
oas$DOB = parse_date_time(oas$date_of_birth_month_year, orders = c("ymd", "dmy", "mdy"))
oas$DOB = gsub(" UTC", "", oas$DOB)
oas$DOB = gsub("2065", "1965", oas$DOB)
oas$date_of_diagnosis_2 = parse_date_time(oas$assessment_date, orders = c("ymd", "dmy", "mdy"))
oas$date_of_diagnosis_2 = gsub(" UTC", "", oas$date_of_diagnosis_2)
oas$age = round(time_length(difftime(as.Date(oas$date_of_diagnosis_2), as.Date(oas$DOB)), "years"))
oas$DOB = NULL
oas$date_of_diagnosis_2 = NULL
oas$hla_b27 = as.character(oas$hla_b27)
oas$hla = recode(oas$hla_b27, "Positive" = "Positive",  "positive" = "Positive", "POSITIVE " = "Positive", "POSITIVE" = "Positive", "Positive ('rare' homozygote)" = "Positive", "Negative" = "Negative", "negative" = "Negative", "NEGATIVE" = "Negative", "negative " = "Negative", "NEGATIVE " = "Negative", "Not done/Not known" = "Not done/Not known", "Not relevant" = "Not done/Not known", "Not relevant " = "Not done/Not known", .default = "Not done/Not known")



#Metadata exclude
list_oas = c("Sample", "Cohort", "drug", "esr", "batch", "crp", "disease_duration_at_time_o", "diagnosis", "smoking_current_ever_never", "ethnicity", "bmi", "gender", "iamc_id", "age", "haemoglobin", "albumin", "alt", "alkaline_phosphatase", "bilirubin_umol")
oas_subset = oas[ , which(colnames(oas) %in% list_oas)]
names(oas_subset) = c("iamc_id", "gender",  "bmi", "ethnicity","smoking_current_ever_never", "disease_duration", "esr", "crp", "haemoglobin", "albumin", "alt", "alkaline_phosphatase", "bilirubin_umol", "Sample", "Cohort", "diagnosis", "drug", "batch", "age")

list_bhm = c("Sample", "Cohort", "drug", "esr", "batch", "crp", "disease_duration_at_time", "arthritis_type", "smoking_current_ever_never", "ethnicity", "bmi", "gender", "iamc_id", "age", "haemoglobin", "albumin", "alt", "alkaline_phosphatase", "bilirubin_umol")
bhm_subset = bhm[ , which(colnames(bhm) %in% list_bhm)]
names(bhm_subset) = c("iamc_id", "gender",  "bmi", "ethnicity","smoking_current_ever_never", "disease_duration", "esr", "crp", "haemoglobin", "albumin", "alt", "alkaline_phosphatase", "bilirubin_umol", "Sample", "Cohort", "batch",  "drug", "diagnosis", "age")

meta_dat = rbind(bhm_subset, oas_subset)
meta_dat$diagnosis = recode(meta_dat$diagnosis, "AS" = "AS", "Clinically Suspect Arthralgia" = "CSA", "Crystal Arthritis" = "Crystal Arthritis", "HC" = "HC", "Healthy Control" = "HC", "Non-Inflammatory Joint Pain" = "NIJP", "NR" = "NR", "other" = "other", "PsA" = "PsA", "RA" = "RA", "Unclassified Inflammatory Arthritis" = "Unclass", "Unknown" = "Unknown", .default = "NA")

#Get the quartiles for crp
summary(meta_dat$crp)
#split the data into the quartiles
quantile(meta_dat$crp, probs = c(0.33, 0.66), na.rm = T)
meta_dat$inflammation = cut(meta_dat$crp, c(0,4,10,245), labels = c("Not inflamed", "Some inflammation", "Inflammation"))
meta_dat$inflammation = ifelse(is.na(as.character(meta_dat$inflammation)),"Not recorded", as.character(meta_dat$inflammation))
meta_dat$inflammation = ifelse(meta_dat$inflammation == "Not recorded" & meta_dat$diagnosis == "HC", "Not inflamed", as.character(meta_dat$inflammation))
meta_dat$crp_corrected = ifelse(meta_dat$crp > 100, "100", as.character(meta_dat$crp))

#Change the data types in some columns to make them more accurate with variables (i.e. smoking is current in 1 or 0 formats so change that column to character instead of int.)
str(meta_dat)
meta_dat$ethnicity = as.character(meta_dat$ethnicity)
meta_dat$smoking_current_ever_never = as.character(meta_dat$smoking_current_ever_never)

summary(meta_dat$esr)
#meta_dat$esr_stat = ifelse(is.na(meta_dat$esr) & meta_dat$diagnosis == "HC", "7", as.character(meta_dat$esr))
#meta_dat$esr_stat = as.numeric(meta_dat$esr_stat)
meta_dat$disease_duration = ifelse(is.na(meta_dat$disease_duration) & meta_dat$diagnosis == "HC", "0", as.character(meta_dat$disease_duration))
meta_dat$disease_duration = as.numeric(meta_dat$disease_duration)
str(meta_dat)

#Deal with the mess that is ethnicity classifications in redcap
meta_dat$ethnicity = recode(meta_dat$ethnicity, "1" = "Not stated", "2" = "White", "3" = "White", "4" = "White", "5" = "Not White", "6" = "Not White", "7" = "Not White", "8" = "Not White", "9" = "Not White", "10" = "Not White", "11" = "Not White", "12" = "Not White", "13" = "Not White", "14" = "Not White", "15" = "Not White", "16" = "Not White", "17" = "Not White", "18" = "White", "19" = "Not White", "20" = "Not White", "21" = "Not White", "22" = "Not White", .default = "Not stated")
meta_dat$ethnicity = ifelse(is.na(meta_dat$ethnicity), "Not stated", as.character(meta_dat$ethnicity))

meta_dat$esr = NULL
meta_dat$bmi = NULL

#hemoglobin : categorical 135 g/L in Men and 120 g/L in women
#atl : categorical  10-40 units per liter (U/L) of blood for men and 7-35 U/L for women

# taxonomic profiles relative abundance
dat_taxa = read.csv("~/Dropbox (Huttenhower Lab)/hutlab/Kelsey/IAMC/Input_data/current_merged_1_11_metaphlan_table.tsv",header=TRUE,row.names=1,sep="\t") / 100
# DNA pathway relative abundance
dat_dna_path = read.csv("~/Dropbox (Huttenhower Lab)/hutlab/Kelsey/IAMC/Input_data/current_merged_1_11_pathway_relab.tsv",header=TRUE,row.names=1,sep="\t")
# DNA pathway relative abundance
dat_dna_ecs = read.csv("~/Desktop/current_merged_1_11_ecs_relab_rename.tsv",sep="\t",row.names = 1, header=T)

### unify ID ###
#At this point this data set does not have the addition of Sample_DNASample
#colnames(dat_taxa) = gsub("Sample_DNASample","",colnames(dat_taxa))
colnames(dat_taxa) = gsub("_taxonomic_profile","",colnames(dat_taxa))
dat_taxa$current_merged = NULL
colnames(dat_dna_path) = gsub("_Abundance","",colnames(dat_dna_path))
colnames(dat_dna_ecs) = gsub("_Abundance.RPKs","",colnames(dat_dna_ecs))


#Exclude samples with little to no reads
sample_exclude = c("BHM00026", "BHM00034", "BHM00041", "BHM00109")
dim(dat_taxa)
dat_taxa = dat_taxa[,-which(colnames(dat_taxa) %in% sample_exclude)]
dim(dat_taxa)

tmp.ind = grep( "\\|c__", rownames(dat_taxa), invert = T )
tmp = dat_taxa[tmp.ind,]
tmp.ind = grep( "\\|p__", rownames(tmp))
dat_phylum = tmp[tmp.ind,]
row.names(dat_phylum) = gsub("k__Bacteria\\|", "", row.names(dat_phylum))

dat_phylum = dat_phylum %>% group_by(row.names(dat_phylum)) %>% summarise_all(list(sum))
colSums(dat_phylum[, -1])
dat_phylum = data.frame(dat_phylum)
row.names(dat_phylum) = dat_phylum$row.names.dat_phylum.
dat_phylum$row.names.dat_phylum. = NULL
dat_phylum = data.frame(t(dat_phylum))
#Check to make sure all samples included have relative abundance
check = colSums(dat_taxa) > 0
check = subset(check, check == FALSE)
check

dat_meta = meta_dat[rownames(meta_dat) %in% colnames(dat_taxa),]
dat_meta = subset(dat_meta, diagnosis != "Unknown" & diagnosis != "NR" & diagnosis != "Crystal Arthritis" & diagnosis != "Unclass" & diagnosis != "other" & diagnosis != "CSA")
dat_taxa = dat_taxa[,colnames(dat_taxa) %in% rownames(dat_meta)]
dat_dna_path = dat_dna_path[,which(colnames(dat_dna_path) %in% colnames(dat_taxa))]
dat_dna_ecs = dat_dna_ecs[,which(colnames(dat_dna_ecs) %in% colnames(dat_taxa))]

dat_meta = dat_meta[match(row.names(dat_meta), names(dat_taxa)), ]

#Create a smaller file to run stats aganist
str(dat_meta)
stat_meta = dat_meta
stat_meta$crp = NULL
stat_meta$Sample = NULL
str(stat_meta)
row.names(stat_meta) = stat_meta$iamc_id
stat_meta$iamc_id = NULL




### select taxa ###
# keep species (without unclassified)
dat_taxa_species = keep_species(dat_taxa)
# select common species (>0.0001 in more than 10% samples)
dat_taxa_species_common = dat_taxa_species[is_common(dat_taxa_species),]
# keep top 25 species (according to median)
dat_taxa_species_top = dat_taxa_species[top_abun(dat_taxa_species,num=10),]

### select DNA pathway ###
# keep unstratified dna pathways
dat_dna_path_unstratified = keep_pathways(dat_dna_path)
dat_dna_ecs_unstratified = keep_pathways(dat_dna_ecs)
# select top 50 unstratified dna pathways (according to median)
dat_dna_path_unstratified_top = dat_dna_path_unstratified[top_abun(dat_dna_path_unstratified,num=10),]
dat_dna_ecs_unstratified_top = dat_dna_ecs_unstratified[top_abun(dat_dna_ecs_unstratified,num=10),]
# select all the unstratified dna pathways related to the top 25 species
dat_dna_path_unstratified_select = select_pathways(dat_path=dat_dna_path,dat_taxa=dat_taxa_species_top )
#dat_dna_ecs_unstratified_select = select_pathways(dat_ecs=dat_dna_ecs,dat_taxa=dat_taxa_species_top )
# select all the top 50 unstratified dna pathways related to the top 25 species
dat_dna_path_unstratified_select_top = dat_dna_path_unstratified_select[top_abun(dat_dna_path_unstratified_select,num=10),]

dat_dna_path_common = dat_dna_path_unstratified[is_common(dat_dna_path_unstratified), ]
dat_dna_ecs_common = dat_dna_ecs_unstratified[is_common(dat_dna_ecs_unstratified), ]

identical(row.names(stat_meta), names(dat_taxa_species_common))
identical(row.names(stat_meta), names(dat_dna_path_common))
identical(row.names(stat_meta), names(dat_dna_ecs_common))
```

\newpage

```{r meta_dat, include = FALSE}

dat_meta$drug = gsub("Not recorded/Not known", "Not recorded", dat_meta$drug)
dat_meta$diagnosis = ordered(dat_meta$diagnosis, c("HC", "NIJP", "AS", "RA", "PsA"))
dat_meta$inflammation = ordered(dat_meta$inflammation, c("Not inflamed", "Some inflammation", "Inflammation", "Not recorded"))
dat_meta$diagnosis = ordered(dat_meta$diagnosis, c("HC", "NIJP", "AS", "PsA", "RA"))
dat_meta$crp_corrected =  as.numeric(dat_meta$crp_corrected)
dat_meta$smoking = recode(dat_meta$smoking_current_ever_never, "1" = "Current", "2" = "Ever", "3" = "Never", "-1" = "Not recorded", .default = "Not recorded")
dat_meta$smoking = as.character(dat_meta$smoking)
dat_meta$smoking[is.na(dat_meta$smoking)] = "Not recorded"
col_smoke = c("Current" = "#DC143C", "Ever" = "#FF7F50", "Never" = "#66CDAA", "Not recorded" = "#696969")

bhm_dat = bhm[row.names(bhm) %in% row.names(dat_meta), ]
bhm_dat$inflammation = dat_meta[match(row.names(bhm_dat), row.names(dat_meta)), "inflammation"]
bhm_dat$anti_ccp_status = recode(bhm_dat$anti_ccp_status, "1" = "Negative", "2" = "Positive", "-1" = "Not recorded", .default = "Not recorded")
bhm_dat$anti_ccp_status[is.na(bhm_dat$anti_ccp_status)] = "Not recorded"
bhm_dat$anti_ccp_status = ordered(bhm_dat$anti_ccp_status, c("Positive", "Negative", "Not recorded"))
bhm_dat$rf = recode(bhm_dat$rf_status, "1" = "Negative", "2" = "Positive", "-1" = "Not recorded", .default = "Not recorded")
bhm_dat$rf[is.na(bhm_dat$rf)] = "Not recorded"
bhm_dat$arthritis_type = droplevels(as.factor(bhm_dat$arthritis_type))
bhm_dat$arthritis_type = as.character(bhm_dat$arthritis_type)
bhm_dat$arthritis_type = mgsub(bhm_dat$arthritis_type, c("Healthy Control", "Non-Inflammatory Joint Pain"), c("HC", "NIJP"))
bhm_dat$rf = ordered(bhm_dat$rf, c("Positive", "Negative", "Not recorded"))

oas_dat = oas[row.names(oas) %in% row.names(dat_meta), ]
oas_dat$inflammation = dat_meta[match(row.names(oas_dat), row.names(dat_meta)), "inflammation"]
oas_dat$hla_b27 = as.factor(oas_dat$hla_b27)
oas_dat$hla_b27 = droplevels(oas_dat$hla_b27)
oas_dat$hla_b27 = gsub(" $", "", oas_dat$hla_b27)
oas_dat$hla = recode(oas_dat$hla_b27, "negative" = "Negative", "Negative" = "Negative", "NEGATIVE " = "Negative", "Not done/Not known" = "Not recorded", "Not relevant" = "Not recorded", "postive" = "Positive", "Positive" = "Positive", "POSITIVE" = "Positive", "Positive ('rare' homozygote)" = "Positive", .default = "Not recorded")
oas_dat$hla = ordered(oas_dat$hla, c("Positive", "Negative", "Not recorded"))

```


\newpage
# Statistics 
Data from HUMAnN2 results:
  \newline
files = 
  \newline
DNA relative abundance: pathabundance_relab.tsv
\newline
*For unstratified pathways (removing the taxa-stratified rows).

## Encoded Pathways  

### Beta diversity  

```{r beta_tax, include= FALSE}
stat_meta = stat_meta[complete.cases(stat_meta$age), ]

stat_meta$smoking = recode(stat_meta$smoking_current_ever_never, "1" = "Current", "2" = "Ever", "3" = "Never", "-1" = "Not recorded", .default = "Not recorded")
stat_meta$smoking = as.character(stat_meta$smoking)
stat_meta$smoking[is.na(stat_meta$smoking)] = "Not recorded"
stat_meta$smoking_current_ever_never = NULL

colSums(is.na(stat_meta))

stat_meta$drug = gsub("Not recorded/Not known", "Not recorded", stat_meta$drug)

meta_disease_duration = stat_meta[complete.cases(stat_meta$disease_duration), ]

meta_crp = stat_meta[complete.cases(stat_meta$crp_corrected), ]
meta_crp$crp_corrected = as.numeric(meta_crp$crp_corrected)


meta_alt = stat_meta[complete.cases(stat_meta$alt), ]
list_alt = c("haemoglobin", "albumin", "alt", "alkaline_phosphatase", "bilirubin_umol", "batch")
meta_alt = meta_alt[, which(names(meta_alt) %in% list_alt)]

na = c("disease_duration", "crp_corrected", "haemoglobin", "albumin", "alt", "alkaline_phosphatase", "bilirubin_umol")
meta_wo_na = stat_meta[, -which(names(stat_meta) %in% na)]
str(meta_wo_na)

```

```{r beta_path, include= FALSE}
#DNA pathway abundance bray
# calculate bray curtis dissimilarity
dat_dna_path_common = data.frame(t(dat_dna_path_common), check.names = F)
identical(row.names(stat_meta), row.names(dat_dna_path_common))
dat_dna_path_common = dat_dna_path_common[row.names(dat_dna_path_common) %in% row.names(stat_meta), ]
bray_path_dna = vegdist(dat_dna_path_common, "bray")
bray_path_dna


dna_disease_duration = dat_dna_path_common[row.names(dat_dna_path_common) %in% row.names(meta_disease_duration), ]
bray_disease_duration = vegdist(dna_disease_duration, "bray")

dna_crp = dat_dna_path_common[row.names(dat_dna_path_common) %in% row.names(meta_crp), ]
bray_crp = vegdist(dna_crp, "bray")

dna_alt = dat_dna_path_common[row.names(dat_dna_path_common) %in% row.names(meta_alt), ]
bray_alt = vegdist(dna_alt, "bray")

```

### Univariate statistics  

```{r adonis_univariate_path, echo = FALSE, warning = FALSE, message = FALSE}
adonis_res_rsq = vector()
adonis_res_pval = vector()
for (col in names(meta_wo_na)){
  adonis.univ = adonis(as.formula(paste("bray_path_dna ~ batch +", col)), data = meta_wo_na)
  adonis_res_rsq[col] = adonis.univ$aov.tab[2,]$R2
  adonis_res_pval[col] = adonis.univ$aov.tab[2,]$`Pr(>F)`
}

disease_duration = adonis(bray_disease_duration ~ batch + disease_duration, data = meta_disease_duration)
adonis_res_pval$disease_duration = disease_duration$aov.tab[2,]$`Pr(>F)`
adonis_res_rsq$disease_duration = disease_duration$aov.tab[2,]$R2

crp_stat = adonis(bray_crp ~ batch + crp_corrected, data = meta_crp)
adonis_res_pval$crp_stat = crp_stat$aov.tab[2,]$`Pr(>F)`
adonis_res_rsq$crp_stat = crp_stat$aov.tab[2,]$R2

for (col in names(meta_alt)){
  adonis.univ = adonis(as.formula(paste("bray_alt ~ batch +", col)), data = meta_alt)
  adonis_res_rsq[col] = adonis.univ$aov.tab[2,]$R2
  adonis_res_pval[col] = adonis.univ$aov.tab[2,]$`Pr(>F)`
}

collect_correct = adonis(bray_path_dna ~ batch + diagnosis + Cohort, data = meta_wo_na)
adonis_res_pval$collect_correct = collect_correct$aov.tab[3,]$`Pr(>F)`
adonis_res_rsq$collect_correct = collect_correct$aov.tab[3,]$R2

gender_correct = adonis(bray_path_dna ~ batch + diagnosis + gender, data = meta_wo_na)
adonis_res_pval$gender_correct = gender_correct$aov.tab[3,]$`Pr(>F)`
adonis_res_rsq$gender_correct = gender_correct$aov.tab[3,]$R2

batch = adonis(bray_path_dna ~ batch, data = meta_wo_na)
adonis_res_pval$batch_real = batch$aov.tab[1,]$`Pr(>F)`
adonis_res_rsq$batch_real = batch$aov.tab[1,]$R2


univar_res_path = rbind(adonis_res_pval, adonis_res_rsq)
univar_res_path = as.data.frame(t(univar_res_path))
names(univar_res_path) = c("P-Value", "R2")
univar_res_path$`P-Value` = as.numeric(univar_res_path$`P-Value`)
univar_res_path$R2 = as.numeric(univar_res_path$R2)
univar_res_path$p_adj = p.adjust(univar_res_path$`P-Value`, "fdr")
univar_res_path = subset(univar_res_path, row.names(univar_res_path) != "batch" & row.names(univar_res_path) != "Cohort" & row.names(univar_res_path) != "gender")
list_rownames = c("Ethnicity", "Current Arthritis Modifying Drug", "Diagnosis", "Age", "Inflammation (crp-value)", "Smoking status", "Disease Duration", "CRP-value", "Haemoglobin", "Albumin", "Alanine Aminotransferase", "Alkaline Phosphatase", "Bilirubin", "Collection Site*", "Sex*", "Sequencing Batch")
univar_res_path$clinical = list_rownames

univar_res_path$stars = cut(univar_res_path$p_adj, c(0, 0.001, 0.01, 0.05, 0.1, Inf), labels = c("***", "**", "*", "'", ""))
rm(adonis_res_pval)
rm(adonis_res_rsq)
```

\newpage 
Univarate statistics of each covariate adjusted by batch (ie model = bray distance ~ batch + covariate). All NA's have been removed per covariate.  

```{r barplot_univ_statistics_all_path, echo= FALSE, warning = FALSE, message = FALSE, dpi = 300}
univar_res_path$p_adj = round(univar_res_path$p_adj, 3)
univar_res_path$R2 = univar_res_path$R2 *100

dodge = position_dodge(width = 0.8)
ggplot(data = univar_res_path, aes(reorder(clinical, univar_res_path$R2), y = R2, label = univar_res_path$p_adj)) + geom_bar(stat = "identity", position = dodge, fill = "#800000", color = "black") + geom_text(position = dodge, vjust = 0.5, hjust = -0.1, size = 3) + theme_bw(base_size = 10) + ylab("Univarate R-squared") + coord_flip() + ylim(0,5) + xlab("")

univar_res_path$data_type = c("Pathways")
```

\newpage

### Pairwise statistics 

Pairwise comparisons of the beta diversity between the different diagnosis  

```{r pairwise_diagnosis_path, echo= FALSE, warning = FALSE, message = FALSE}
bray_mat = as.matrix(bray_path_dna)
path_pair_dia = pairwise.adonis(bray_mat, factors = stat_meta$diagnosis, p.adjust.m = "BH")

path_pair_dia %>%
  kbl() %>%
  kable_minimal()
```

\newpage

Pairwise comparisons of the beta diversity between the different inflammation statuses  


```{r pairwise_inflammtion_path, echo= FALSE, warning = FALSE, message = FALSE}
path_pair_inflam = pairwise.adonis(bray_mat, factors = stat_meta$inflammation, p.adjust.m = "BH")

path_pair_inflam %>%
  kbl() %>%
  kable_minimal()

```

\newpage

Pairwise comparisons of the beta diversity between the different sequencing batches  


```{r pairwise_batch_path, echo= FALSE, warning = FALSE, message = FALSE}
path_pair_batch = pairwise.adonis(bray_mat, factors = stat_meta$batch, p.adjust.m = "BH")

path_pair_batch %>%
  kbl() %>%
  kable_minimal()

```

\newpage

### Comparisons to HC populations

Boxplots comparisons of the bray distance between samples. All groups are compared to the control samples. Here they are colored by diagnosis.

```{r bray_comparisons_path_diangosis, echo= FALSE, warning = FALSE, message = FALSE, dpi = 300}

bray_df = melt(bray_mat)
bray_df = bray_df[bray_df$Var1 != bray_df$Var2, ]
bray_df = cbind(dat_meta[ match(bray_df$Var1, row.names(dat_meta)), "diagnosis"], bray_df)
names(bray_df) = c("diagnosis_row", "row", "col", "bray_distance")
bray_df = cbind(dat_meta[ match(bray_df$col, row.names(dat_meta)), "diagnosis"], bray_df)
names(bray_df) = c("diagnosis_col", "diagnosis_row", "row", "col", "bray_distance")

df_HC_dia_path = subset(bray_df, diagnosis_row == "HC")
df_HC_dia_path$comparisons = paste(df_HC_dia_path$diagnosis_col, df_HC_dia_path$diagnosis_row, sep = "_")

ggplot(df_HC_dia_path, aes(x = diagnosis_col, y = bray_distance, color = diagnosis_col, add = "jitter")) + geom_boxplot(lwd = 2) + theme_bw(base_size = 10) + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) + xlab("Average change") + ylab("Bray Distance") + scale_color_manual(values = col_diagnosis) + labs(color = "Patient diagnosis")

df_HC_dia_path$data_type = "Pathways"

```

\newpage

Boxplots comparisons of the bray distance between samples. All groups are compared to the control samples. Here they are colored by inflammation.  

```{r bray_comparisons_path_inflammation, echo= FALSE, warning = FALSE, message = FALSE, dpi = 300}

bray_df = melt(bray_mat)
bray_df = bray_df[bray_df$Var1 != bray_df$Var2, ]
bray_df = cbind(dat_meta[ match(bray_df$Var1, row.names(dat_meta)), "inflammation"], bray_df)
names(bray_df) = c("inflammation_row", "row", "col", "bray_distance")
bray_df = cbind(dat_meta[ match(bray_df$col, row.names(dat_meta)), "inflammation"], bray_df)
names(bray_df) = c("inflammation_col", "inflammation_row", "row", "col", "bray_distance")



df_HC_inflam_path = subset(bray_df, inflammation_row == "Not inflamed")
df_HC_inflam_path$comparisons = paste(df_HC_inflam_path$inflammation_col, df_HC_inflam_path$inflammation_row, sep = "_")
df_HC_inflam_path$inflammation_col = ordered(df_HC_inflam_path$inflammation_col, c("Not inflamed", "Some inflammation", "Inflammation", "Not recorded"))

ggplot(df_HC_inflam_path, aes(x = inflammation_col, y = bray_distance, color = inflammation_col, add = "jitter")) + geom_boxplot(lwd = 2) + theme_bw(base_size = 10) + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) + xlab("Average change") + ylab("Bray Distance") + scale_color_manual(values = col_inflammation) + labs(color = "Inflammation (CRP)")

df_HC_inflam_path$data_type = "Pathways"

```

\newpage

## DNA encoded enzyme categories  

```{r beta_ecs, include= FALSE}
#DNA pathway abundance bray
# calculate bray curtis dissimilarity
dat_dna_ecs_common = t(dat_dna_ecs_common)
identical(row.names(stat_meta), row.names(dat_dna_ecs_common))
dat_dna_ecs_common = dat_dna_ecs_common[row.names(dat_dna_ecs_common) %in% row.names(stat_meta), ]
bray_ecs_dna = vegdist(dat_dna_ecs_common, "bray")
bray_ecs_dna

dna_disease_duration = dat_dna_ecs_common[row.names(dat_dna_ecs_common) %in% row.names(meta_disease_duration), ]
bray_disease_duration = vegdist(dna_disease_duration, "bray")

dna_crp = dat_dna_ecs_common[row.names(dat_dna_ecs_common) %in% row.names(meta_crp), ]
bray_crp = vegdist(dna_crp, "bray")

dna_alt = dat_dna_ecs_common[row.names(dat_dna_ecs_common) %in% row.names(meta_alt), ]
bray_alt = vegdist(dna_alt, "bray")

```


```{r adonis_univariate_ecs, echo = FALSE, warning = FALSE, message = FALSE}
adonis_res_rsq = vector()
adonis_res_pval = vector()
for (col in names(meta_wo_na)){
  adonis.univ = adonis(as.formula(paste("bray_ecs_dna ~ batch +", col)), data = meta_wo_na)
  adonis_res_rsq[col] = adonis.univ$aov.tab[2,]$R2
  adonis_res_pval[col] = adonis.univ$aov.tab[2,]$`Pr(>F)`
}

disease_duration = adonis(bray_disease_duration ~ batch + disease_duration, data = meta_disease_duration)
adonis_res_pval$disease_duration = disease_duration$aov.tab[2,]$`Pr(>F)`
adonis_res_rsq$disease_duration = disease_duration$aov.tab[2,]$R2

crp_stat = adonis(bray_crp ~ batch + crp_corrected, data = meta_crp)
adonis_res_pval$crp_stat = crp_stat$aov.tab[2,]$`Pr(>F)`
adonis_res_rsq$crp_stat = crp_stat$aov.tab[2,]$R2

for (col in names(meta_alt)){
  adonis.univ = adonis(as.formula(paste("bray_alt ~ batch +", col)), data = meta_alt)
  adonis_res_rsq[col] = adonis.univ$aov.tab[2,]$R2
  adonis_res_pval[col] = adonis.univ$aov.tab[2,]$`Pr(>F)`
}

collect_correct = adonis(bray_ecs_dna ~ batch + diagnosis + Cohort, data = meta_wo_na)
adonis_res_pval$collect_correct = collect_correct$aov.tab[3,]$`Pr(>F)`
adonis_res_rsq$collect_correct = collect_correct$aov.tab[3,]$R2

gender_correct = adonis(bray_ecs_dna ~ batch + diagnosis + gender, data = meta_wo_na)
adonis_res_pval$gender_correct = gender_correct$aov.tab[3,]$`Pr(>F)`
adonis_res_rsq$gender_correct = gender_correct$aov.tab[3,]$R2

batch = adonis(bray_ecs_dna ~ batch, data = meta_wo_na)
adonis_res_pval$batch_real = batch$aov.tab[1,]$`Pr(>F)`
adonis_res_rsq$batch_real = batch$aov.tab[1,]$R2


univar_res_ecs = rbind(adonis_res_pval, adonis_res_rsq)
univar_res_ecs = as.data.frame(t(univar_res_ecs))
names(univar_res_ecs) = c("P-Value", "R2")
univar_res_ecs$`P-Value` = as.numeric(univar_res_ecs$`P-Value`)
univar_res_ecs$R2 = as.numeric(univar_res_ecs$R2)
univar_res_ecs$p_adj = p.adjust(univar_res_ecs$`P-Value`, "fdr")
univar_res_ecs = subset(univar_res_ecs, row.names(univar_res_ecs) != "batch" & row.names(univar_res_ecs) != "Cohort" & row.names(univar_res_ecs) != "gender")
list_rownames = c("Ethnicity", "Current Arthritis Modifying Drug", "Diagnosis", "Age", "Inflammation (crp-value)", "Smoking status", "Disease Duration", "CRP-value", "Haemoglobin", "Albumin", "Alanine Aminotransferase", "Alkaline Phosphatase", "Bilirubin", "Collection Site*", "Sex*", "Sequencing Batch")
univar_res_ecs$clinical = list_rownames

univar_res_ecs$stars = cut(univar_res_ecs$p_adj, c(0, 0.001, 0.01, 0.05, 0.1, Inf), labels = c("***", "**", "*", "'", ""))
rm(adonis_res_pval)
rm(adonis_res_rsq)
```

### Univariate statistics   

Univarate statistics of each covariate adjusted by batch (ie model = bray distance ~ batch + covariate). All NA's have been removed per covariate.    
```{r barplot_univ_statistics_ecs, echo= FALSE, warning = FALSE, message = FALSE, dpi = 300}
univar_res_ecs$p_adj = round(univar_res_ecs$p_adj, 3)
univar_res_ecs$R2 = univar_res_ecs$R2 *100

dodge = position_dodge(width = 0.8)
ggplot(data = univar_res_ecs, aes(reorder(clinical, univar_res_ecs$R2), y = R2, label = univar_res_ecs$p_adj)) + geom_bar(stat = "identity", position = dodge, fill = "#800000", color = "black") + geom_text(position = dodge, vjust = 0.5, hjust = -0.1, size = 3) + theme_bw(base_size = 10) + ylab("Univarate R-squared") + coord_flip() + ylim(0,5) + xlab("")

univar_res_ecs$data_type = c("ECs")
```

\newpage

### Pairwise statistics   

Pairwise comparisons of the beta diversity between the different diagnosis  

```{r pairwise_diagnosis_ecs, echo= FALSE, warning = FALSE, message = FALSE}
bray_mat = as.matrix(bray_ecs_dna)
ecs_pair_dia = pairwise.adonis(bray_mat, factors = stat_meta$diagnosis, p.adjust.m = "BH")

ecs_pair_dia %>%
  kbl() %>%
  kable_minimal()
```

\newpage

Pairwise comparisons of the beta diversity between the different inflammation categories    


```{r pairwise_inflammation_ecs, echo= FALSE, warning = FALSE, message = FALSE}
ecs_pair_inflam = pairwise.adonis(bray_mat, factors = stat_meta$inflammation, p.adjust.m = "BH")

ecs_pair_inflam  %>%
  kbl() %>%
  kable_minimal()
```

\newpage

### Pairwise comparisons to HC populations  

Boxplots comparisons of the bray distance between samples. All groups are compared to the control samples. Here they are colored by diagnosis.  

```{r bray_comparisons_ecs_diangosis, echo= FALSE, warning = FALSE, message = FALSE, dpi = 300}

bray_df = melt(bray_mat)
bray_df = bray_df[bray_df$Var1 != bray_df$Var2, ]
bray_df = cbind(dat_meta[ match(bray_df$Var1, row.names(dat_meta)), "diagnosis"], bray_df)
names(bray_df) = c("diagnosis_row", "row", "col", "bray_distance")
bray_df = cbind(dat_meta[ match(bray_df$col, row.names(dat_meta)), "diagnosis"], bray_df)
names(bray_df) = c("diagnosis_col", "diagnosis_row", "row", "col", "bray_distance")



df_HC_dia_ecs = subset(bray_df, diagnosis_row == "HC")
df_HC_dia_ecs$comparisons = paste(df_HC_dia_ecs$diagnosis_col, df_HC_dia_ecs$diagnosis_row, sep = "_")

ggplot(df_HC_dia_ecs, aes(x = diagnosis_col, y = bray_distance, color = diagnosis_col, add = "jitter")) + geom_boxplot(lwd = 2) + theme_bw(base_size = 10) + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) + xlab("Average change") + ylab("Bray Distance") + scale_color_manual(values = col_diagnosis) + labs(color = "Patient diagnosis")

df_HC_dia_ecs$data_type = "ECs"

```

\newpage

Boxplots comparisons of the bray distance between samples. All groups are compared to the control samples. Here they are colored by inflammation.  

```{r bray_comparisons_ecs_inflammation, echo= FALSE, warning = FALSE, message = FALSE, dpi = 300}

bray_df = melt(bray_mat)
bray_df = bray_df[bray_df$Var1 != bray_df$Var2, ]
bray_df = cbind(dat_meta[ match(bray_df$Var1, row.names(dat_meta)), "inflammation"], bray_df)
names(bray_df) = c("inflammation_row", "row", "col", "bray_distance")
bray_df = cbind(dat_meta[ match(bray_df$col, row.names(dat_meta)), "inflammation"], bray_df)
names(bray_df) = c("inflammation_col", "inflammation_row", "row", "col", "bray_distance")



df_HC_inflam_ecs = subset(bray_df, inflammation_row == "Not inflamed")
df_HC_inflam_ecs$comparisons = paste(df_HC_inflam_ecs$inflammation_col, df_HC_inflam_ecs$inflammation_row, sep = "_")
df_HC_inflam_ecs$inflammation_col = ordered(df_HC_inflam_ecs$inflammation_col, c("Not inflamed", "Some inflammation", "Inflammation", "Not recorded"))

ggplot(df_HC_inflam_ecs, aes(x = inflammation_col, y = bray_distance, color = inflammation_col, add = "jitter")) + geom_boxplot(lwd = 2) + theme_bw(base_size = 10) + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) + xlab("Average change") + ylab("Bray Distance") + scale_color_manual(values = col_inflammation) + labs(color = "Inflammation (CRP)")

df_HC_inflam_ecs$data_type = "ECs"

```


\newpage

# Basic Vis 

Visualizations of Beta Diversity DNA Encoded Pathways   

```{r pcoa_path_diagnosis, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
pc = capscale(bray_path_dna ~1, comm = dat_dna_path_common)
cap = data.frame(pc$CA$u)
cap = merge(stat_meta, cap, by = "row.names")
rownames(cap) = cap[,1]
cap = cap[,2:ncol(cap)]
s = summary(pc)

dat_taxa_species_common = data.frame(t(dat_taxa_species_common), check.names = F)
dat_taxa_species_common = dat_taxa_species_common[row.names(dat_taxa_species_common) %in% row.names(stat_meta), ]

data_wa = data.frame(wascores(pc$CA$u[, 1:2], dat_taxa_species_common))
data_wa_sub = data_wa[which(abs(data_wa$MDS1)>0.045 | abs(data_wa$MDS2)>0.045),]

ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = diagnosis), size = 2, alpha = 0.7) + theme_bw(base_size = 16)  + scale_color_manual(values = col_diagnosis) + scale_alpha_manual(guide='none', values = list(a = 0.2, point = 1)) + labs(color = "Patient diagnosis") + coord_fixed() + theme(legend.position = "NONE") +
  labs(title="",
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))

```


\newpage
Colored by inflammation with species overlay.  

```{r pcoa_path_inflammation, echo = FALSE, warning = FALSE, message = FALSE, fig.height= 5, fig.width=6.5, dpi = 600}
library(shadowtext)
library(ggExtra)
cap$inflammation = ordered(cap$inflammation, c("Not inflamed", "Some inflammation", "Inflammation", "Not recorded"))
row.names(data_wa_sub) = gsub("s__", "", row.names(data_wa_sub))
row.names(data_wa_sub) = gsub("_", " ", row.names(data_wa_sub))
data_wa_sub = subset(data_wa_sub, row.names(data_wa_sub) != "Clostridium clostridioforme")

grob = grobTree(textGrob("PERMANOVA: R-sq  = 0.02, FDR p-val = 0.007", x = 0.58, y = 0.98, gp=gpar(col="black", fontsize=9)))
#png("PCOA_inflammation_manuscript.png", height = 5, width = 6.5, units = "in", res = 600)
pcoa_plot = ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = inflammation), size = 2, alpha = 0.65) + 
  theme_bw(base_size = 12) + scale_color_manual(values = col_inflammation) +  coord_fixed() + geom_shadowtext(data=data_wa_sub, aes(x = data_wa_sub$MDS1, y = data_wa_sub$MDS2, label = rownames(data_wa_sub)), check_overlap = TRUE, vjust = 0, hjust = 0.57, size = 3, fontface = "italic",color = "black", bg.color = "white") + labs(color = "Inflammation (CRP)") +  theme(legend.position = "NONE") + annotation_custom(grob) +
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
#png("PCOA_inflammation_density_manuscript.png", height = 5, width = 6.5, units = "in", res = 600)
pcoa_density = ggExtra::ggMarginal(pcoa_plot, size = 7, colour = NA, groupFill = TRUE, groupColour = T)
print(pcoa_density)
#dev.off()
```

\newpage

Colored by hemoglobin  


```{r pcoa_hemo_plot_path, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
#png("PCOA_hemoglobin.png", height = 4, width = 5, units = "in", res = 600)
ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = haemoglobin), size = 3, alpha = 0.7) + 
  theme_bw(base_size = 12) + scale_color_viridis(option = "D")  + coord_fixed() + labs(color = "Hemoglobin") + 
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
```


\newpage

Colored by the relative abundance of the phylum Firmicutes in each sample.  

```{r Firm_plot_path, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
cap$Firmicutes = dat_phylum[match(row.names(cap), row.names(dat_phylum)), "p__Firmicutes"]
cap$Bacteroidetes = dat_phylum[match(row.names(cap), row.names(dat_phylum)), "p__Bacteroidetes"]

#png("PCOA_jia_Firm.png", height = 10, width = 10, units = "in", res = 600)
ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = Firmicutes), size = 3, alpha = 0.7) + 
  theme_bw() + scale_color_viridis(option = "D")  + coord_fixed(1) +
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
```

\newpage

Colored by the relative abundance of the phylum Bacteroidetes in each sample.   

```{r Bac_plot_path, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
#png("PCOA_jia_Bacteroidetes.png", height = 10, width = 10, units = "in", res = 600)
ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = Bacteroidetes), size = 3, alpha = 0.7) + 
  theme_bw() + scale_color_viridis(option = "D")  + coord_fixed() +
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
```

\newpage

## DNA encoded ECs  

```{r pcoa_ecs_diagnosis, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
pc = capscale(bray_ecs_dna ~1, comm = dat_dna_ecs_common)
cap = data.frame(pc$CA$u)
cap = merge(stat_meta, cap, by = "row.names")
rownames(cap) = cap[,1]
cap = cap[,2:ncol(cap)]
s = summary(pc)

data_wa = data.frame(wascores(pc$CA$u[, 1:2], dat_taxa_species_common))
data_wa_sub = data_wa[which(abs(data_wa$MDS1)>0.045 | abs(data_wa$MDS2)>0.045),]

ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = diagnosis), size = 2, alpha = 0.7) + theme_bw(base_size = 16)  + scale_color_manual(values = col_diagnosis) + scale_alpha_manual(guide='none', values = list(a = 0.2, point = 1)) + labs(color = "Patient diagnosis") + coord_fixed() + theme(legend.position = "NONE") +
  labs(title="",
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#pdf("PCoA_cross_poster_diagnosis.pdf",width=7,height=6)
#print(p)
#dev.off()

```


\newpage
Colored by inflammation with species overlay.  

```{r pcoa_ecs_inflammation, echo = FALSE, warning = FALSE, message = FALSE, fig.height= 5, fig.width=6.5, dpi = 600}
library(shadowtext)
library(ggExtra)
cap$inflammation = ordered(cap$inflammation, c("Not inflamed", "Some inflammation", "Inflammation", "Not recorded"))
row.names(data_wa_sub) = gsub("s__", "", row.names(data_wa_sub))
row.names(data_wa_sub) = gsub("_", " ", row.names(data_wa_sub))
data_wa_sub = subset(data_wa_sub, row.names(data_wa_sub) != "Clostridium clostridioforme")

grob = grobTree(textGrob("PERMANOVA: R-sq  = 0.014, FDR p-val = 0.003", x = 0.58, y = 0.98, gp=gpar(col="black", fontsize=9)))
#png("PCOA_inflammation_manuscript.png", height = 5, width = 6.5, units = "in", res = 600)
pcoa_plot = ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = inflammation), size = 2, alpha = 0.65) + 
  theme_bw(base_size = 12) + scale_color_manual(values = col_inflammation) +  coord_fixed() + geom_shadowtext(data=data_wa_sub, aes(x = data_wa_sub$MDS1, y = data_wa_sub$MDS2, label = rownames(data_wa_sub)), check_overlap = TRUE, vjust = 0, hjust = 0.57, size = 3, fontface = "italic",color = "black", bg.color = "white") + labs(color = "Inflammation (CRP)") +  theme(legend.position = "NONE") + annotation_custom(grob) +
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
#png("PCOA_inflammation_density_manuscript.png", height = 5, width = 6.5, units = "in", res = 600)
pcoa_density = ggExtra::ggMarginal(pcoa_plot, size = 7, colour = NA, groupFill = TRUE, groupColour = T)
print(pcoa_density)
#dev.off()
```

\newpage

Colored by hemoglobin  


```{r pcoa_hemo_plot_ecs, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
#png("PCOA_hemoglobin.png", height = 4, width = 5, units = "in", res = 600)
ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = haemoglobin), size = 3, alpha = 0.7) + 
  theme_bw(base_size = 12) + scale_color_viridis(option = "D")  + coord_fixed() + labs(color = "Hemoglobin") + 
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
```


\newpage

Colored by the realtive abundance of the phylum Firmicutes in each sample.  

```{r Firm_plot_ecs, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
cap$Firmicutes = dat_phylum[match(row.names(cap), row.names(dat_phylum)), "p__Firmicutes"]
cap$Bacteroidetes = dat_phylum[match(row.names(cap), row.names(dat_phylum)), "p__Bacteroidetes"]

#png("PCOA_jia_Firm.png", height = 10, width = 10, units = "in", res = 600)
ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = Firmicutes), size = 3, alpha = 0.7) + 
  theme_bw() + scale_color_viridis(option = "D")  + coord_fixed(1) +
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
```
\newpage

Colored by the realtive abundance of the phylum Bacteroidetes in each sample.   

```{r Bac_plot_ecs, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}
#png("PCOA_jia_Bacteroidetes.png", height = 10, width = 10, units = "in", res = 600)
ggplot(cap, aes(MDS1, MDS2)) + geom_point(aes(color = Bacteroidetes), size = 3, alpha = 0.7) + 
  theme_bw() + scale_color_viridis(option = "D")  + coord_fixed() +
  labs(title="", 
       x = paste("Axis 1 (", round(s$cont$importance[2,1]*100, digits =2), "%)", sep = ''),
       y = paste("Axis 2 (", round(s$cont$importance[2,2]*100, digits =2), "%)", sep = ''))
#dev.off()
```

\newpage

# Per-feature difference 

Model = _feature ~ Age + diagnosis.endpoint_  
Subset to just the Vitamin B pathways  

``` {r maaslin2_heatmap_function_VB, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 10, fig.height = 12, dpi = 300}

maas_path_dia = read.csv("~/Dropbox (Harvard University)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional/Figures/Figure_4/MaAsLin2_Functions/maaslin2_path_diagnosis_wo_tech/significant_results.tsv", sep = "\t")
maas_path_dia = subset(maas_path_dia, value == "RA" | value == "AS")

maas_path_inflam = read.csv("~/Dropbox (Harvard University)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional/Figures/Figure_4/MaAsLin2_Functions/maaslin2_path_inflam_wo_tech/significant_results.tsv", sep = "\t")
maas_path_inflam = subset(maas_path_inflam, value == "Inflammation" | value == "Some ")
maas_path_inflam$value = gsub("Some ", "Some inflammation", maas_path_inflam$value)

maas_path_hem = read.csv("~/Dropbox (Harvard University)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional/Figures/Figure_4/MaAsLin2_Functions/maaslin2_path_hemo_cat_wo_tech/significant_results.tsv", sep = "\t")
maas_path_hem = subset(maas_path_hem, value == "Normal")
maas_path_hem$coef = maas_path_hem$coef * -1
maas_path_hem$value = gsub("Normal", "Anemia", maas_path_hem$value)

maas_ecs_dia = read.csv("~/Dropbox (Harvard University)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional/Figures/Figure_4/MaAsLin2_Functions/maaslin2_ecs_diagnosis_wo_tech/significant_results.tsv", sep = "\t")
maas_ecs_dia = subset(maas_ecs_dia, value == "RA" | value == "AS")

maas_ecs_inflam = read.csv("~/Dropbox (Harvard University)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional/Figures/Figure_4/MaAsLin2_Functions/maaslin2_ecs_inflam_wo_tech/significant_results.tsv", sep = "\t")
maas_ecs_inflam = subset(maas_ecs_inflam, value == "Inflammation" | value == "Some ")
maas_ecs_inflam$value = gsub("Some ", "Some inflammation", maas_ecs_inflam$value)

maas_ecs_hem = read.csv("~/Dropbox (Harvard University)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional/Figures/Figure_4/MaAsLin2_Functions/maaslin2_ecs_hemo_cat_wo_tech/significant_results.tsv", sep = "\t")
maas_ecs_hem = subset(maas_ecs_hem, value == "Normal")
maas_ecs_hem$coef = maas_ecs_hem$coef * -1
maas_ecs_hem$value = gsub("Normal", "Anemia", maas_ecs_hem$value)


maas_sign = rbind(maas_path_dia, maas_path_hem, maas_path_inflam, maas_ecs_dia, maas_ecs_hem, maas_ecs_inflam)
maas_sign = subset(maas_sign, metadata != "age")
maas_sign$value = droplevels(maas_sign$value)

maas_sign$cor = -log(maas_sign$qval)*sign(maas_sign$coef)
maas_sign$stars = cut(maas_sign$qval, c(0, 0.001, 0.01, 0.05, 0.1, Inf), labels = c("***", "**", "*", "'", ""))

cat = read.csv("~/Dropbox (Harvard University)/hutlab/Kelsey/IAMC/Manuscript1_cross-sectional/Figures/Figure_4/Final_full_function.csv")
cat$Category = gsub("  $", "", cat$Category)
cat$Category = gsub(" $", "", cat$Category)

cat$Specifics = gsub("  $", "", cat$Specifics)
cat$Specifics = gsub(" $", "", cat$Specifics)

path_names = data.frame(row.names(dat_dna_path_unstratified))
names(path_names) = "Function"
row.names(path_names) = path_names$Function
path_names = data.frame(t(path_names))
path_names = data.frame(t(path_names))
path_names$function2 = row.names(path_names)

ecs_names = data.frame(row.names(dat_dna_ecs_unstratified))
names(ecs_names) = "Function"
row.names(ecs_names) = ecs_names$Function
ecs_names = data.frame(t(ecs_names))
ecs_names = data.frame(t(ecs_names))
ecs_names$function2 = gsub(":.*", "", ecs_names$Function)

func_list = rbind(path_names, ecs_names)
func_list$tname = row.names(func_list)
cat$name = func_list[match(cat$Function, func_list$function2), "Function"]
cat$reduce = func_list[match(cat$Function, func_list$function2), "tname"]


maas_sign$names = cat[match(maas_sign$feature,  cat$reduce), "name"]
maas_sign = maas_sign[maas_sign$feature %in% cat$reduce, ]
maas_sign$cat = cat[match(maas_sign$feature, cat$reduce), "Category"]
maas_sign$spec = cat[match(maas_sign$feature, cat$reduce), "Specifics"]

maas_sign_man_vit = subset(maas_sign, cat == "Cofactor, carrier, and vitamin biosynthesis")

maas_sign_man_vit = subset(maas_sign_man_vit, spec != "Vitamin K" & spec != "Reductant biosynthesis" & spec != "Carrier biosynthesis" & spec != "Cofactor")

maas_sign_man_vit$spec2 = mgsub(maas_sign_man_vit$spec, c("Vitamin B", "Vitamin B12", "Vitamin B2", "Vitamin B7", "Vitamin B1", "Vitamin B9", "Vitamin B6"), c("B", "B12", "B2", "B7", "B1", "B9", "B6"))


maas_sign_man_vit$names = gsub(".*\\: ", "", maas_sign_man_vit$names)
maas_sign_man_vit$names = gsub("2.6.1.85", "Aminodeoxychorismate synthase", maas_sign_man_vit$names)
maas_sign_man_vit$names = gsub("5.4.99.61", "Precorrin-8X methylmutase", maas_sign_man_vit$names)


ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_sign_man_vit) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-5, 10), breaks = c(-6, -3, 0, 3, 6, 9)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(spec2 ~ ., scales = "free", space = "free")
```

\newpage

Subset of vitmain B's -->  9/12    

``` {r maaslin2_heatmap_function_VB_Fig4, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 14, fig.height = 10, dpi = 600}
maas_sign_man_vit_sub = subset(maas_sign_man_vit, spec2 == "B9" | spec2 == "B12")
maas_sign_man_vit_sub = subset(maas_sign_man_vit_sub, value != "Some inflammation")
maas_sign_man_vit_sub$spec2 = factor(maas_sign_man_vit_sub$spec2, levels = c("B9", "B12"))
#png("MaAsLin2_heatmap_function_vitaminb.png", height = 12, width = 12, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_sign_man_vit_sub) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-5, 7), breaks = c(-5, -3, 0, 3, 6)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 18) + theme(legend.position="bottom") + facet_grid(spec2 ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage 

Other cofactor/vitamin pathways    

``` {r maaslin2_heatmap_function_cofactor, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 12, dpi = 300}

maas_sign_man_vit = subset(maas_sign, cat == "Cofactor, carrier, and vitamin biosynthesis")

maas_sign_man_vit = subset(maas_sign_man_vit, spec == "Vitamin K" | spec == "Reductant biosynthesis" | spec == "Carrier biosynthesis" | spec == "Cofactor")
maas_sign_man_vit$spec = gsub("Reductant biosynthesis", "Reductant", maas_sign_man_vit$spec)

maas_sign_man_vit$names = gsub(".*\\: ", "", maas_sign_man_vit$names)
#png("MaAsLin2_heatmap_function_vitaminb.png", height = 12, width = 12, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_sign_man_vit) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-5.2, 10), breaks = c(-5, -2, 0, 2, 5, 10)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(spec ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage

Iron sequestration   

``` {r maaslin2_heatmap_manuscript_iron_fig5, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 14, fig.height = 8, dpi = 600}
maas_sign_man_iron = subset(maas_sign, cat == "Iron sequestration" | cat == "Phytate")
maas_sign_man_iron$spec = mgsub(maas_sign_man_iron$spec, c("Bacterial non-heme ferritin", "ABC transporter", "Ferritin", "Heme", "Heme transporter", "Non-heme iron", "Siderophore and metallophore biosynthesis", "Tetrapyrrole Biosynthesis", "Phytate Biosynthesis", "Phytate Degradation"), c("Ferritin", "ABC", "Ferritin", "Heme", "Heme", "Ferritin", "Siderophore", "Heme", "Phytate", "Phytate"))
maas_sign_man_iron = subset(maas_sign_man_iron, value != "Some inflammation")
maas_sign_man_iron$names = gsub(".*\\: ", "", maas_sign_man_iron$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_sign_man_iron) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-6, 10), breaks = c(-6, -3, 0, 3, 6, 9)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 18) + theme(legend.position="bottom") + facet_grid(spec ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage 

Supplementary Carbs  

``` {r maaslin2_heatmap_supplement_carb, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 15, dpi = 300}
maas_carb = subset(maas_sign, cat == "Carbohydrate biosynthesis" | cat == "Carbohydrate degradation" | cat == "Carboxylate degradation")
maas_carb$spec2 = mgsub(maas_carb$spec, c("CO2 Fixation", "CoA-biosynthesis", "Colanic acid", "Lipopolysaccharides", "Polysaccharide Degradation", "Propanoate"), c("other", "other", "other", "other", "other", "other"))
maas_carb$spec2[maas_carb$spec2 == ""] = "other"

maas_carb$names = gsub(".*\\: ", "", maas_carb$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_carb) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-9.1, 10.4), breaks = c(-10, -5, 0, 5, 11)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(spec2 ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage

Supplementary Amino acid   

``` {r maaslin2_heatmap_supplement_amino, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 17, dpi = 300}
maas_amino = subset(maas_sign, cat == "Amino acid biosynthesis" | cat == "Amino acid degradation")
maas_amino$spec2 = mgsub(maas_amino$spec, c("Arginine/ornithine/proline", "Asparagine", "Asparate/asparagine", "Cirulline", "Homocycteine", "Homocycteine and methionine", "Homoserine", "Lysine/threonine/methionine", "Proline", "Selenocysteine", "Serine", "Thiamine", "Threonine", "Tryosine", "Valine", "Phenylalanine", "Glutamine", "Tyrosine"), c("other", "other", "other", "other", "Cysteine",  "Cysteine", "other", "other", "other", "Cysteine", "other", "other", "other", "other", "other", "other", "other", "other"))
maas_amino$spec2[maas_amino$spec2 == ""] = "other"

maas_amino$names = gsub(".*\\: ", "", maas_amino$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_amino) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-9, 12.6), breaks = c(-9, -6, -3, 0, 3, 6, 9, 12)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(spec2 ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage

Supplementary Nucleotide   
 
``` {r maaslin2_heatmap_supplement_nuc, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 9, dpi = 300}
maas_nuc = subset(maas_sign, cat == "Nucleoside and nucleotide degradation" | cat == "Nucleoside and nucleotide biosynthesis")
maas_nuc$extra = gsub("Nucleoside and nucleotide ", "", maas_nuc$cat)
maas_nuc$spec2 = paste(maas_nuc$spec, maas_nuc$extra, sep = " ")

maas_nuc$names = gsub(".*\\: ", "", maas_nuc$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_nuc) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-10.9, 8.6), breaks = c(-10, -5, 0, 4, 8)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(spec2 ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage

Supplementary Fatty Acid     

``` {r maaslin2_heatmap_supplement_fatty, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 13, dpi = 300}
maas_fatty = subset(maas_sign, cat == "Fatty acid and lipid biosynthesis" | cat == "Fatty acid degradation" | cat == "Fermentation to short-chain fatty acid")
maas_fatty$cat = gsub("Fatty acid degradation", "Degradation", maas_fatty$cat)

maas_fatty$names = gsub(".*\\: ", "", maas_fatty$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_fatty) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-9.1, 9.8), breaks = c(-10, -5, 0, 5, 10)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(cat ~ ., scales = "free", space = "free")
#dev.off()
```


\newpage

Supplementary Terpenoid     

``` {r maaslin2_heatmap_supplement_terpenoid, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 6, dpi = 300}
maas_iso = subset(maas_sign, cat == "Terpenoid")
maas_iso$spec = mgsub(maas_iso$spec, c("IsoIsopentenyl Diphosphate Biosynthesis", "Methylerythritol Phosphate Pathways", "Mevalonate Pathways"), c("other", "other", "Mevalonate"))
maas_iso$spec[maas_iso$spec == ""] = "other"

maas_iso$names = gsub(".*\\: ", "", maas_iso$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_iso) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-9.2, 10.6), breaks = c(-10, -5, 0, 5, 10)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(spec ~ ., scales = "free", space = "free")
#dev.off()
```


\newpage

Supplementary Aromatic     

``` {r maaslin2_heatmap_supplement_aromatic, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 6, dpi = 300}
maas_aro = subset(maas_sign, cat == "Amine and Polyamine" | cat == "Aromatic")

maas_aro$names = gsub(".*\\: ", "", maas_aro$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_aro) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-7.1, 9.6), breaks = c(-8, -4, 0, 5, 10)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(cat ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage

Supplementary Environmental response      

``` {r maaslin2_heatmap_supplement_environ, echo = FALSE, warning = FALSE, message = FALSE, fig.width = 12, fig.height = 7, dpi = 300}
maas_env = subset(maas_sign, cat == "Cell wall" | cat == "Environmental response" | cat == "ABC transporter")
maas_env$cat = gsub("ABC transporter", "ABC", maas_env$cat)

maas_env$names = gsub(".*\\: ", "", maas_env$names)
#png("MaAsLin2_heatmap_iron.png", height = 6, width = 10, res = 600, units = "in")
ggplot(aes(y = reorder(names, cor), x= value, fill= cor), data= maas_env) + geom_tile() + scale_fill_gradient2(low="#2C7BB6", mid="white", high="#D7191C", limits = c(-10.4, 8.2), breaks = c(-10, -5, 0, 4, 8)) + geom_text(aes(label=stars), color="black", size=4) + labs(y=NULL, x=NULL, fill="-log(qval) * sign(Coef)") + theme_classic(base_size = 12) + theme(legend.position="bottom") + facet_grid(cat ~ ., scales = "free", space = "free")
#dev.off()
```

\newpage

# randomForest 
Random Forest analysis on arcsin(pathways)  
Set with a root of 123 and 1000 trees.  

## Pathways  

```{r setup_all, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
### 07/22/19 Started the RF analysis ###
str(stat_meta)
##Attempting this compare the HC to those patients with inflammation

rf_meta = stat_meta[, c(1,4:8, 10:14, 16)]
rf_meta = subset(rf_meta, inflammation == "Not inflamed" | inflammation == "Inflammation")
rf_meta = rf_meta[rf_meta$diagnosis == "HC" & rf_meta$inflammation == "Not inflamed" | rf_meta$inflammation == "Inflammation", ]
#rf_meta = rf_meta[complete.cases(rf_meta),]

rf_meta_bugs = rf_meta[, -c(2:9)]

```

```{r setup_randomforest_path, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

rf_meta_bugs[, c(1,3:4)] = lapply(rf_meta_bugs[, c(1,3:4)], as.factor)

#species_rf = asinh(dat_taxa_species_common)
AST <- function(x) {
    return(sign(x) * asin(sqrt(abs(x))))
}

species_rf = apply(dat_dna_path_common, 2, AST)
species_rf = data.frame(species_rf)
species_rf = species_rf[row.names(species_rf) %in% row.names(rf_meta), ]
dat_rf = merge(rf_meta_bugs, species_rf, by = "row.names")
row.names(dat_rf) = dat_rf$Row.names
dat_rf$Row.names = NULL


# CROSS-VALIDATE 5-fold to determine number of predictors for random forest model
rf_crossval = rfcv(dat_rf, dat_rf[,3], step = 0.8 )
# check how error increases with decreased number of predictors
rf_crossval$error.cv
with(rf_crossval, plot(n.var, error.cv, log="x", type="o", lwd=2)) # 50? predictors

# separate data into testing and training sets.
set.seed(123)
samp <- sample(nrow(dat_rf), 0.8 * nrow(dat_rf))
train <- dat_rf[samp, ]
test <- dat_rf[-samp, ]

# build the basic model, using 111 predictors based on cross-validation above
names(train)
basic_model_path = randomForest(inflammation ~ ., data = train, ntree = 1000, importance = T, proximity = T, confusion = T, err.rate = T, nodesize = 1)

### Importance 
imp_path = importance(basic_model_path)
imp_path = data.frame(predictors = rownames(imp_path), imp_path)
imp_path = arrange(imp_path, desc(MeanDecreaseGini))
imp_path$predictors = factor(imp_path$predictors, levels = imp_path$predictors)

# Select the top 50 predictors
imp_path_top = imp_path[1:15, ]

```

```{r model_information_path, echo = FALSE, warning = FALSE, message = FALSE}
basic_model_path
```

\newpage

### Top predictors 

```{r important_gini_path, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300, fig.height=5, fig.width= 10}
ggplot(imp_path_top, aes(x = predictors, y = MeanDecreaseGini)) +
  geom_bar(stat = "identity", fill = "#5F9EA0") +
  coord_flip() + xlab("Predictors") + ylab("Mean decrease gini score") +
  ggtitle("") + theme_bw(base_size = 10) 
```

\newpage 

Accuracy of predictors

```{r important_accuracy_path, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300, fig.height=5, fig.width= 10}
ggplot(imp_path_top, aes(x = predictors, y = MeanDecreaseAccuracy)) +
  geom_bar(stat = "identity", fill = "#6A5ACD") +
  coord_flip() + xlab("Predictors") + ylab("Mean decrease accuracy") +
  ggtitle("") + theme_bw(base_size = 10) 
```

\newpage
ROC curves     

```{r basic_roc_path, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}

pred_test_path = predict(basic_model_path, test, index=2, type="prob", norm.votes=TRUE, predict.all=FALSE, proximity=FALSE, nodes=FALSE)

pred_test_path <- data.frame(pred_test_path)
library(pROC)
pred_test_roc_path <- roc(test$inflammation, pred_test_path$Not.inflamed)
#auc(pred_test_roc)

#plot(pred_test_roc)
ggroc(pred_test_roc_path, lwd=1.2, col="blue")+
geom_abline(intercept = 1, slope = 1, color = "red", linetype = "dashed", lwd=1.2) + theme_bw()


```

\newpage

## ECs  

Random Forest analysis on arcsin(ECs)  
Set with a root of 123 and 1000 trees.   


```{r setup_randomforest_ecs, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#species_rf = asinh(dat_taxa_species_common)
AST <- function(x) {
    return(sign(x) * asin(sqrt(abs(x))))
}

species_rf = apply(dat_dna_ecs_common, 2, AST)
species_rf = data.frame(species_rf)
species_rf = species_rf[row.names(species_rf) %in% row.names(rf_meta), ]
dat_rf = merge(rf_meta_bugs, species_rf, by = "row.names")
row.names(dat_rf) = dat_rf$Row.names
dat_rf$Row.names = NULL


# CROSS-VALIDATE 5-fold to determine number of predictors for random forest model
rf_crossval = rfcv(dat_rf, dat_rf[,3], step = 0.8 )
# check how error increases with decreased number of predictors
rf_crossval$error.cv
with(rf_crossval, plot(n.var, error.cv, log="x", type="o", lwd=2)) # 50? predictors

# separate data into testing and training sets.
set.seed(123)
samp <- sample(nrow(dat_rf), 0.8 * nrow(dat_rf))
train <- dat_rf[samp, ]
test <- dat_rf[-samp, ]

# build the basic model, using 111 predictors based on cross-validation above
names(train)
basic_model_ecs = randomForest(inflammation ~ ., data = train, ntree = 1000, importance = T, proximity = T, confusion = T, err.rate = T, nodesize = 1)

### Importance 
imp_ecs = importance(basic_model_ecs)
imp_ecs = data.frame(predictors = rownames(imp_ecs), imp_ecs)
imp_ecs = arrange(imp_ecs, desc(MeanDecreaseGini))
imp_ecs$predictors = factor(imp_ecs$predictors, levels = imp_ecs$predictors)

# Select the top 50 predictors
imp_ecs_top = imp_ecs[1:20, ]

```

```{r model_information_ecs, echo = FALSE, warning = FALSE, message = FALSE}
basic_model_ecs
```

\newpage

### Top predictors 

```{r important_gini_ecs, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300, fig.height=5, fig.width= 10}
ggplot(imp_ecs_top, aes(x = predictors, y = MeanDecreaseGini)) +
  geom_bar(stat = "identity", fill = "#5F9EA0") +
  coord_flip() + xlab("Predictors") + ylab("Mean decrease gini score") +
  ggtitle("") + theme_bw(base_size = 10) 
```

\newpage 

Accuracy of predictors

```{r important_accuracy_ecs, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300, fig.height=5, fig.width= 10}
ggplot(imp_ecs_top, aes(x = predictors, y = MeanDecreaseAccuracy)) +
  geom_bar(stat = "identity", fill = "#6A5ACD") +
  coord_flip() + xlab("Predictors") + ylab("Mean decrease accuracy") +
  ggtitle("") + theme_bw(base_size = 10) 
```

\newpage
ROC curves     

```{r basic_roc_ecs, echo = FALSE, warning = FALSE, message = FALSE, dpi = 300}

pred_test_ecs = predict(basic_model_ecs, test, index=2, type="prob", norm.votes=TRUE, predict.all=FALSE, proximity=FALSE, nodes=FALSE)

pred_test_ecs = data.frame(pred_test_ecs)
library(pROC)
pred_test_roc_ecs = roc(test$inflammation, pred_test_ecs$Not.inflamed)
#auc(pred_test_roc)

#plot(pred_test_roc)
ggroc(pred_test_roc_ecs, lwd=1.2, col="blue")+
geom_abline(intercept = 1, slope = 1, color = "red", linetype = "dashed", lwd=1.2) + theme_bw()


```

\newpage

# Per-feature

## Vitamin B9 (Folate)  

1CMET2-PWY: N10-formyl-tetrahydrofolate biosynthesis  

```{r n-10_formyl_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

formyl = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`1CMET2-PWY: N10-formyl-tetrahydrofolate biosynthesis`)
names(formyl) = c("Samples", "rel_abun")

nonzero_formyl = subset(formyl, rel_abun != 0)
nonzero_formyl$diagnosis = stat_meta[match(nonzero_formyl$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_formyl$inflammation = stat_meta[match(nonzero_formyl$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_formyl, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_formyl$rel_abun = log10(nonzero_formyl$rel_abun/median_value)


nonzero_formyl$diagnosis = factor(nonzero_formyl$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_formyl_final = subset(nonzero_formyl, diagnosis != "NIJP" & diagnosis != "PsA")

#png("formyl_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_formyl = ggplot(nonzero_formyl_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.5, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "N10-formyl-tetrahydrofolate biosynthesis") + ylab("Density") + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_formyl)
#dev.off()
```

\newpage

### Pathway breakdown  

1CMET2-PWY: N10-formyl-tetrahydrofolate biosynthesis

```{r composition_tax_1cmetspwy, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("1CMET2-PWY: N10-formyl-tetrahydrofolate biosynthesis", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`1CMET2-PWY: N10-formyl-tetrahydrofolate biosynthesis` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("1CMET2-PWY: N10-formyl-tetrahydrofolate biosynthesis\\|g__.*\\.", "", names(bar))
names(bar) = gsub("1CMET2-PWY: N10-formyl-tetrahydrofolate biosynthesis\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_n10-formyl, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="N10-formyl-tetrahydrofolate biosynthesis") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 


```{r n-10_formyl_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_formyl$inflammation = factor(nonzero_formyl$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_formyl = subset(nonzero_formyl, inflammation != "Not recorded")
#png("formyl_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_formyl_inflamm = ggplot(nonzero_formyl, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.5, .5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "N10-formyl-tetrahydrofolate biosynthesis") + ylab("Density") + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_formyl_inflamm) 
#dev.off()

```

\newpage

folate transformations II       

```{r folate_transform_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

folate = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-3841: folate transformations II`)
names(folate) = c("Samples", "rel_abun")

nonzero_folate = subset(folate, rel_abun != 0)
nonzero_folate$diagnosis = stat_meta[match(nonzero_folate$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_folate$inflammation = stat_meta[match(nonzero_folate$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_folate, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_folate$rel_abun = log10(nonzero_folate$rel_abun/median_value)


nonzero_folate$diagnosis = factor(nonzero_folate$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_folate_final = subset(nonzero_folate, diagnosis != "NIJP" & diagnosis != "PsA")

#png("folate_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_folate = ggplot(nonzero_folate_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.5, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Folate transformations II") + ylab("Density") + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_folate)
#dev.off()
```

\newpage

### Pathway breakdown  

PWY-3841: folate transformations II

```{r composition_tax_pwy3841, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-3841: folate transformations II", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-3841: folate transformations II` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-3841: folate transformations II\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-3841: folate transformations II\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_folate_transform, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Folate transformations II") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r folate_transform_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_folate$inflammation = factor(nonzero_folate$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_folate = subset(nonzero_folate, inflammation != "Not recorded")
#png("folate_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_folate_inflamm = ggplot(nonzero_folate, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.5, .5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Folate transformations II") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank())  + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_folate_inflamm)
#dev.off()

```
 
 
\newpage 
 
S-adenosyl-L-methionine cycle I      

```{r adenosyl_meth_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

adenosyl = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-6151: S-adenosyl-L-methionine cycle I`)
names(adenosyl) = c("Samples", "rel_abun")

nonzero_adenosyl = subset(adenosyl, rel_abun != 0)
nonzero_adenosyl$diagnosis = stat_meta[match(nonzero_adenosyl$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_adenosyl$inflammation = stat_meta[match(nonzero_adenosyl$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_adenosyl, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_adenosyl$rel_abun = log10(nonzero_adenosyl$rel_abun/median_value)


nonzero_adenosyl$diagnosis = factor(nonzero_adenosyl$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_adenosyl_final = subset(nonzero_adenosyl, diagnosis != "NIJP" & diagnosis != "PsA")

#png("adenosyl_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_adenosyl = ggplot(nonzero_adenosyl_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.5, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "S-adenosyl-L-methionine cycle I") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_adenosyl)
#dev.off()
```

\newpage

### Pathway breakdown  

PWY-6151: S-adenosyl-L-methionine cycle I

```{r composition_tax_pwy6151, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-6151: S-adenosyl-L-methionine cycle I", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-6151: S-adenosyl-L-methionine cycle I` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-6151: S-adenosyl-L-methionine cycle I\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-6151: S-adenosyl-L-methionine cycle I\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_adenosyl_meth, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="S-adenosyl-L-methionine cycle I") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


```{r adenosyl_meth_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_adenosyl$inflammation = factor(nonzero_adenosyl$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_adenosyl = subset(nonzero_adenosyl, inflammation != "Not recorded")
#png("adenosyl_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_adenosyl_inflamm = ggplot(nonzero_adenosyl, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.5, .5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "S-adenosyl-L-methionine cycle I") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank())  + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_adenosyl_inflamm)
#dev.off()

```

\newpage 

1.5.1.3..Dihydrofolate.reductase     

```{r dihydrofolate_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}
dat_dna_ecs_common = data.frame(dat_dna_ecs_common, check.names = F)
dihydrofolate = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`1.5.1.3: Dihydrofolate reductase`)
names(dihydrofolate) = c("Samples", "rel_abun")

nonzero_dihydrofolate = subset(dihydrofolate, rel_abun != 0)
nonzero_dihydrofolate$diagnosis = stat_meta[match(nonzero_dihydrofolate$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_dihydrofolate$inflammation = stat_meta[match(nonzero_dihydrofolate$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_dihydrofolate, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_dihydrofolate$rel_abun = log10(nonzero_dihydrofolate$rel_abun/median_value)


nonzero_dihydrofolate$diagnosis = factor(nonzero_dihydrofolate$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_dihydrofolate_final = subset(nonzero_dihydrofolate, diagnosis != "NIJP" & diagnosis != "PsA")

#png("dihydrofolate_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_dihydrofolate = ggplot(nonzero_dihydrofolate_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.5, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Dihydrofolate reductase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_dihydrofolate)
#dev.off()
```

\newpage

### Pathway breakdown  

1.5.1.3: Dihydrofolate reductase

```{r composition_tax_1513, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("1.5.1.3: Dihydrofolate reductase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`1.5.1.3: Dihydrofolate reductase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("1.5.1.3: Dihydrofolate reductase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("1.5.1.3: Dihydrofolate reductase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_dihydrofolate, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Dihydrofolate reductase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r dihydrofolate_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_dihydrofolate$inflammation = factor(nonzero_dihydrofolate$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_dihydrofolate = subset(nonzero_dihydrofolate, inflammation != "Not recorded")
#png("dihydrofolate_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_dihydrofolate_inflamm = ggplot(nonzero_dihydrofolate, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.5, .5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Dihydrofolate reductase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_dihydrofolate_inflamm) 
#dev.off()

```

\newpage 

Formyltetrahydrofolate deformylase     

```{r formyl_deformylase_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

deformylase = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`3.5.1.10: Formyltetrahydrofolate deformylase`)
names(deformylase) = c("Samples", "rel_abun")

nonzero_deformylase = subset(deformylase, rel_abun != 0)
nonzero_deformylase$diagnosis = stat_meta[match(nonzero_deformylase$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_deformylase$inflammation = stat_meta[match(nonzero_deformylase$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_deformylase, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_deformylase$rel_abun = log10(nonzero_deformylase$rel_abun/median_value)


nonzero_deformylase$diagnosis = factor(nonzero_deformylase$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_deformylase_final = subset(nonzero_deformylase, diagnosis != "NIJP" & diagnosis != "PsA")

#png("deformylase_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_deformylase = ggplot(nonzero_deformylase_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.2, 1.2) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Formyltetrahydrofolate deformylase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_deformylase)
#dev.off()
```

\newpage

### Pathway breakdown  

3.5.1.10: Formyltetrahydrofolate deformylase

```{r composition_tax_35110, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("3.5.1.10: Formyltetrahydrofolate deformylase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`3.5.1.10: Formyltetrahydrofolate deformylase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("3.5.1.10: Formyltetrahydrofolate deformylase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("3.5.1.10: Formyltetrahydrofolate deformylase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_n-10_foryml_deformylase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Formyltetrahydrofolate deformylase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r n-10_formyl_deformylase_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_deformylase$inflammation = factor(nonzero_deformylase$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_deformylase = subset(nonzero_deformylase, inflammation != "Not recorded")
#png("deformylase_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_deformylase_inflamm = ggplot(nonzero_deformylase, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.25, 1.25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Formyltetrahydrofolate deformylase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_deformylase_inflamm) 
#dev.off()

```

\newpage 

4.1.2.25: Dihydroneopterin aldolase     

```{r aldolase_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

aldolase = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`4.1.2.25: Dihydroneopterin aldolase`)
names(aldolase) = c("Samples", "rel_abun")

nonzero_aldolase = subset(aldolase, rel_abun != 0)
nonzero_aldolase$diagnosis = stat_meta[match(nonzero_aldolase$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_aldolase$inflammation = stat_meta[match(nonzero_aldolase$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_aldolase, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_aldolase$rel_abun = log10(nonzero_aldolase$rel_abun/median_value)


nonzero_aldolase$diagnosis = factor(nonzero_aldolase$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_aldolase_final = subset(nonzero_aldolase, diagnosis != "NIJP" & diagnosis != "PsA")

#png("aldolase_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_aldolase = ggplot(nonzero_aldolase_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.5, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Dihydroneopterin aldolase") + ylab("Density") + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_aldolase)
#dev.off()
```

\newpage

### Pathway breakdown  

4.1.2.25: Dihydroneopterin aldolase

```{r composition_tax_41225, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("4.1.2.25: Dihydroneopterin aldolase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`4.1.2.25: Dihydroneopterin aldolase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("4.1.2.25: Dihydroneopterin aldolase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("4.1.2.25: Dihydroneopterin aldolase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_aldolase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Dihydroneopterin aldolase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r aldolase_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_aldolase$inflammation = factor(nonzero_aldolase$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_aldolase = subset(nonzero_aldolase, inflammation != "Not recorded")
#png("aldolase_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_aldolase_inflamm = ggplot(nonzero_aldolase, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.5, .5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Dihydroneopterin aldolase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_aldolase_inflamm)
#dev.off()

```

\newpage 

5-formyltetrahydrofolate cyclo-ligase        

```{r 5_formyltetra_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

ligase = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`6.3.3.2: 5-formyltetrahydrofolate cyclo-ligase`)
names(ligase) = c("Samples", "rel_abun")

nonzero_ligase = subset(ligase, rel_abun != 0)
nonzero_ligase$diagnosis = stat_meta[match(nonzero_ligase$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_ligase$inflammation = stat_meta[match(nonzero_ligase$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_ligase, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_ligase$rel_abun = log10(nonzero_ligase$rel_abun/median_value)


nonzero_ligase$diagnosis = factor(nonzero_ligase$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_ligase_final = subset(nonzero_ligase, diagnosis != "NIJP" & diagnosis != "PsA")

#png("ligase_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_ligase = ggplot(nonzero_ligase_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.25, .25) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "5-formyltetrahydrofolate cyclo-ligase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_ligase)
#dev.off()
```

\newpage

### Pathway breakdown  

6.3.3.2: 5-formyltetrahydrofolate cyclo-ligase

```{r composition_tax_6332, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("6.3.3.2: 5-formyltetrahydrofolate cyclo-ligase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`6.3.3.2: 5-formyltetrahydrofolate cyclo-ligase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("6.3.3.2: 5-formyltetrahydrofolate cyclo-ligase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("6.3.3.2: 5-formyltetrahydrofolate cyclo-ligase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_5_formyltetra, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="5-formyltetrahydrofolate cyclo-ligase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r 5_formyltetra_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_ligase$inflammation = factor(nonzero_ligase$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_ligase = subset(nonzero_ligase, inflammation != "Not recorded")
#png("ligase_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_ligase_inflamm = ggplot(nonzero_ligase, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.25, .25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "5-formyltetrahydrofolate cyclo-ligase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_ligase_inflamm) 
#dev.off()

```

\newpage 

Glycine hydroxymethyltransferase        

```{r glycine_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

glycine = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.2.1: Glycine hydroxymethyltransferase`)
names(glycine) = c("Samples", "rel_abun")

nonzero_glycine = subset(glycine, rel_abun != 0)
nonzero_glycine$diagnosis = stat_meta[match(nonzero_glycine$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_glycine$inflammation = stat_meta[match(nonzero_glycine$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_glycine, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_glycine$rel_abun = log10(nonzero_glycine$rel_abun/median_value)


nonzero_glycine$diagnosis = factor(nonzero_glycine$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_glycine_final = subset(nonzero_glycine, diagnosis != "NIJP" & diagnosis != "PsA")

#png("glycine_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_glycine = ggplot(nonzero_glycine_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.25, .25) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Glycine hydroxymethyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_glycine)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.2.1: Glycine hydroxymethyltransferase

```{r composition_tax_2121, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.2.1: Glycine hydroxymethyltransferase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.2.1: Glycine hydroxymethyltransferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.2.1: Glycine hydroxymethyltransferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.2.1: Glycine hydroxymethyltransferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_glycine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
 theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Glycine hydroxymethyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r glycine_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_glycine$inflammation = factor(nonzero_glycine$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_glycine = subset(nonzero_glycine, inflammation != "Not recorded")
#png("glycine_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_glycine_inflamm = ggplot(nonzero_glycine, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.25, .25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Glycine hydroxymethyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_glycine_inflamm) 
#dev.off()

```

\newpage 

FADH(2)-oxidizing       

```{r FADH_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

FADH = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.74: (FADH(2)-oxidizing)`)
names(FADH) = c("Samples", "rel_abun")

nonzero_FADH = subset(FADH, rel_abun != 0)
nonzero_FADH$diagnosis = stat_meta[match(nonzero_FADH$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_FADH$inflammation = stat_meta[match(nonzero_FADH$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_FADH, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_FADH$rel_abun = log10(nonzero_FADH$rel_abun/median_value)


nonzero_FADH$diagnosis = factor(nonzero_FADH$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_FADH_final = subset(nonzero_FADH, diagnosis != "NIJP" & diagnosis != "PsA")

#png("FADH_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_FADH = ggplot(nonzero_FADH_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.75, .75) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "FADH(2)-oxidizing") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_FADH)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.1.74: (FADH(2)-oxidizing)

```{r composition_tax_21174, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.74:", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.74: (FADH(2)-oxidizing)` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.74: \\(FADH\\(2)-oxidizing)\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.74: \\(FADH\\(2)-oxidizing)\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_FAD, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="FADH(2)-oxidizing") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r FADH_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_FADH$inflammation = factor(nonzero_FADH$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_FADH = subset(nonzero_FADH, inflammation != "Not recorded")
#png("FADH_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_FADH_inflamm = ggplot(nonzero_FADH, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.75, .75) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "FADH(2)-oxidizing") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_FADH_inflamm)
#dev.off()

```

\newpage 

Uroporphyrinogen-III C-methyltransferase   

```{r uroporphyrinogen_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

uroporphyrinogen = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.107: Uroporphyrinogen-III C-methyltransferase`)
names(uroporphyrinogen) = c("Samples", "rel_abun")

nonzero_uroporphyrinogen = subset(uroporphyrinogen, rel_abun != 0)
nonzero_uroporphyrinogen$diagnosis = stat_meta[match(nonzero_uroporphyrinogen$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_uroporphyrinogen$inflammation = stat_meta[match(nonzero_uroporphyrinogen$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_uroporphyrinogen, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_uroporphyrinogen$rel_abun = log10(nonzero_uroporphyrinogen$rel_abun/median_value)


nonzero_uroporphyrinogen$diagnosis = factor(nonzero_uroporphyrinogen$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_uroporphyrinogen_final = subset(nonzero_uroporphyrinogen, diagnosis != "NIJP" & diagnosis != "PsA")

#png("uroporphyrinogen_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_uroporphyrinogen = ggplot(nonzero_uroporphyrinogen_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.25, 1.25) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Uroporphyrinogen-III C-methyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_uroporphyrinogen)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.1.107: Uroporphyrinogen-III C-methyltransferase

```{r composition_tax_211107, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.107: Uroporphyrinogen-III C-methyltransferase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.107: Uroporphyrinogen-III C-methyltransferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.107: Uroporphyrinogen-III C-methyltransferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.107: Uroporphyrinogen-III C-methyltransferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_uroporphyrinogenIII, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Uroporphyrinogen-III C-methyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


```{r uroporphyrinogen_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_uroporphyrinogen$inflammation = factor(nonzero_uroporphyrinogen$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_uroporphyrinogen = subset(nonzero_uroporphyrinogen, inflammation != "Not recorded")
#png("uroporphyrinogen_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_uroporphyrinogen_inflamm = ggplot(nonzero_uroporphyrinogen, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.25, 1.25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Uroporphyrinogen-III C-methyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_uroporphyrinogen_inflamm) 
#dev.off()

```

\newpage 

Aminodeoxychorismate synthase           

```{r oxychorismate_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

oxychorismate = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`6.3.5.8: Transferred entry: 2.6.1.85`)
names(oxychorismate) = c("Samples", "rel_abun")

nonzero_oxychorismate = subset(oxychorismate, rel_abun != 0)
nonzero_oxychorismate$diagnosis = stat_meta[match(nonzero_oxychorismate$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_oxychorismate$inflammation = stat_meta[match(nonzero_oxychorismate$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_oxychorismate, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_oxychorismate$rel_abun = log10(nonzero_oxychorismate$rel_abun/median_value)


nonzero_oxychorismate$diagnosis = factor(nonzero_oxychorismate$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_oxychorismate_final = subset(nonzero_oxychorismate, diagnosis != "NIJP" & diagnosis != "PsA")

#png("oxychorismate_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_oxychorismate = ggplot(nonzero_oxychorismate_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.75, 1.5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Aminodeoxychorismate synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_oxychorismate)
#dev.off()
```

\newpage

### Pathway breakdown  

6.3.5.8: Transferred entry: 2.6.1.85

```{r composition_tax_6358, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("6.3.5.8: Transferred entry: 2.6.1.85", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`6.3.5.8: Transferred entry: 2.6.1.85` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("6.3.5.8: Transferred entry: 2.6.1.85\\|g__.*\\.", "", names(bar))
names(bar) = gsub("6.3.5.8: Transferred entry: 2.6.1.85\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_oxychorismate, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Aminodeoxychorismate synthase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r oxychorismate_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_oxychorismate$inflammation = factor(nonzero_oxychorismate$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_oxychorismate = subset(nonzero_oxychorismate, inflammation != "Not recorded")
#png("oxychorismate_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_oxychorismate_inflamm = ggplot(nonzero_oxychorismate, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.75, 1.5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Aminodeoxychorismate synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_oxychorismate_inflamm)
#dev.off()

```

\newpage 

Homocysteine S-methyltransferase      

```{r homocysteine_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

homocysteine = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.10: Homocysteine S-methyltransferase`)
names(homocysteine) = c("Samples", "rel_abun")

nonzero_homocysteine = subset(homocysteine, rel_abun != 0)
nonzero_homocysteine$diagnosis = stat_meta[match(nonzero_homocysteine$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_homocysteine$inflammation = stat_meta[match(nonzero_homocysteine$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_homocysteine, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_homocysteine$rel_abun = log10(nonzero_homocysteine$rel_abun/median_value)


nonzero_homocysteine$diagnosis = factor(nonzero_homocysteine$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_homocysteine_final = subset(nonzero_homocysteine, diagnosis != "NIJP" & diagnosis != "PsA")

#png("homocysteine_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_homocysteine = ggplot(nonzero_homocysteine_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-2, 2.5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Homocysteine S-methyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_homocysteine)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.1.10: Homocysteine S-methyltransferase

```{r composition_tax_21110, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.10: Homocysteine S-methyltransferase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.10: Homocysteine S-methyltransferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.10: Homocysteine S-methyltransferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.10: Homocysteine S-methyltransferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))


col = scale_fill_manual(values = c("#3288BD", "#5E4FA2", "#66A61E", "#FFED6F", "#FF0000", "#F46D43", "#E7298A", "#00008B", "#8DD3C7", "#FFFFB3", "#CCCCCC"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_homocysteine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Homocysteine S-methyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r homocysteine_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_homocysteine$inflammation = factor(nonzero_homocysteine$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_homocysteine = subset(nonzero_homocysteine, inflammation != "Not recorded")
#png("homocysteine_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_homocysteine_inflamm = ggplot(nonzero_homocysteine, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 2.5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Homocysteine S-methyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_homocysteine_inflamm)
#dev.off()

```

\newpage 

Dihydropteroate synthase       

```{r dihydropteroate_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

dihydropteroate = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.5.1.15: Dihydropteroate synthase`)
names(dihydropteroate) = c("Samples", "rel_abun")

nonzero_dihydropteroate = subset(dihydropteroate, rel_abun != 0)
nonzero_dihydropteroate$diagnosis = stat_meta[match(nonzero_dihydropteroate$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_dihydropteroate$inflammation = stat_meta[match(nonzero_dihydropteroate$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_dihydropteroate, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_dihydropteroate$rel_abun = log10(nonzero_dihydropteroate$rel_abun/median_value)


nonzero_dihydropteroate$diagnosis = factor(nonzero_dihydropteroate$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_dihydropteroate_final = subset(nonzero_dihydropteroate, diagnosis != "NIJP" & diagnosis != "PsA")

#png("dihydropteroate_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_dihydropteroate = ggplot(nonzero_dihydropteroate_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.5, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Dihydropteroate synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE") 
print(density_dihydropteroate)
#dev.off()
```

\newpage

### Pathway breakdown  

2.5.1.15: Dihydropteroate synthase

```{r composition_tax_25115, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.5.1.15: Dihydropteroate synthase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.5.1.15: Dihydropteroate synthase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.5.1.15: Dihydropteroate synthase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.5.1.15: Dihydropteroate synthase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]

f = data.frame(t(f))
head(f)

f$Taxa <- row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_dihydropteroate, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Dihydropteroate synthase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r dihydropteroate_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_dihydropteroate$inflammation = factor(nonzero_dihydropteroate$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_dihydropteroate = subset(nonzero_dihydropteroate, inflammation != "Not recorded")
#png("dihydropteroate_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_dihydropteroate_inflamm = ggplot(nonzero_dihydropteroate, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-0.5, .5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Dihydropteroate synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_dihydropteroate_inflamm)
#dev.off()

```

\newpage 

Methenyltetrahydrofolate cyclohydrolase      

```{r cyclohydrolase_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

cyclohydrolase = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`3.5.4.9: Methenyltetrahydrofolate cyclohydrolase`)
names(cyclohydrolase) = c("Samples", "rel_abun")

nonzero_cyclohydrolase = subset(cyclohydrolase, rel_abun != 0)
nonzero_cyclohydrolase$diagnosis = stat_meta[match(nonzero_cyclohydrolase$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_cyclohydrolase$inflammation = stat_meta[match(nonzero_cyclohydrolase$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_cyclohydrolase, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_cyclohydrolase$rel_abun = log10(nonzero_cyclohydrolase$rel_abun/median_value)


nonzero_cyclohydrolase$diagnosis = factor(nonzero_cyclohydrolase$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_cyclohydrolase_final = subset(nonzero_cyclohydrolase, diagnosis != "NIJP" & diagnosis != "PsA")

#png("cyclohydrolase_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_cyclohydrolase = ggplot(nonzero_cyclohydrolase_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.75, 1) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Methenyltetrahydrofolate cyclohydrolase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_cyclohydrolase)
#dev.off()
```

\newpage

### Pathway breakdown  

3.5.4.9: Methenyltetrahydrofolate cyclohydrolase

```{r composition_tax_3549, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("3.5.4.9: Methenyltetrahydrofolate cyclohydrolase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`3.5.4.9: Methenyltetrahydrofolate cyclohydrolase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("3.5.4.9: Methenyltetrahydrofolate cyclohydrolase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("3.5.4.9: Methenyltetrahydrofolate cyclohydrolase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_cyclohydrolase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Methenyltetrahydrofolate cyclohydrolase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r cyclohydrolase_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_cyclohydrolase$inflammation = factor(nonzero_cyclohydrolase$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_cyclohydrolase = subset(nonzero_cyclohydrolase, inflammation != "Not recorded")
#png("cyclohydrolase_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_cyclohydrolase_inflamm = ggplot(nonzero_cyclohydrolase, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.75, 1) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Methenyltetrahydrofolate cyclohydrolase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_cyclohydrolase_inflamm)
#dev.off()

```

\newpage 

Methionine synthase    

```{r methionine_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

methionine = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.13: Methionine synthase`)
names(methionine) = c("Samples", "rel_abun")

nonzero_methionine = subset(methionine, rel_abun != 0)
nonzero_methionine$diagnosis = stat_meta[match(nonzero_methionine$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_methionine$inflammation = stat_meta[match(nonzero_methionine$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_methionine, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_methionine$rel_abun = log10(nonzero_methionine$rel_abun/median_value)


nonzero_methionine$diagnosis = factor(nonzero_methionine$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_methionine_final = subset(nonzero_methionine, diagnosis != "NIJP" & diagnosis != "PsA")

#png("methionine_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_methionine = ggplot(nonzero_methionine_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Methionine synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_methionine)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.1.13: Methionine synthase

```{r composition_tax_21113, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.13: Methionine synthase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.13: Methionine synthase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.13: Methionine synthase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.13: Methionine synthase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_methionine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Methionine synthase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

```{r methionine_density_inflammation, echo = FALSE, warning = FALSE, message = FALSE}
nonzero_methionine$inflammation = factor(nonzero_methionine$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_methionine = subset(nonzero_methionine, inflammation != "Not recorded")
#png("methionine_denisty_inflammation.png", height = 3, width = 7, units = "in", res = 600)
density_methionine_inflamm = ggplot(nonzero_methionine, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1, .5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Methionine synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_methionine_inflamm)
#dev.off()

```

\newpage  

## Viatmin B12  

COBALSYN.PWY..adenosylcobalamin.salvage.from.cobinamide.I - hemo_cat   

```{r COBALSYN.PWY_density_hem, echo = FALSE, warning = FALSE, message = FALSE}
stat_meta$haemo_cat = as.character(stat_meta$haemoglobin)
stat_meta$haemo_cat =  ifelse(is.na(stat_meta$haemo_cat) == FALSE, "Normal", as.character(stat_meta$haemo_cat))
stat_meta$haemo_cat =  ifelse(stat_meta$gender ==  "m" & stat_meta$haemoglobin < 135, "Anemia", as.character(stat_meta$haemo_cat))
stat_meta$haemo_cat =  ifelse(stat_meta$gender ==  "f" & stat_meta$haemoglobin < 120, "Anemia", as.character(stat_meta$haemo_cat))

COBALSYN.PWY = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`COBALSYN-PWY: adenosylcobalamin salvage from cobinamide I`)
names(COBALSYN.PWY) = c("Samples", "rel_abun")

nonzero_COBALSYN.PWY = subset(COBALSYN.PWY, rel_abun != 0)
nonzero_COBALSYN.PWY$hem = stat_meta[match(nonzero_COBALSYN.PWY$Samples, row.names(stat_meta)), "haemo_cat"]
tmp = subset(nonzero_COBALSYN.PWY, hem == "Normal")
median_value = median(tmp$rel_abun)
nonzero_COBALSYN.PWY$rel_abun = log10(nonzero_COBALSYN.PWY$rel_abun/median_value)


nonzero_COBALSYN.PWY$hem = factor(nonzero_COBALSYN.PWY$hem, levels = c("Normal", "Anemia", "Not recorded"))
nonzero_COBALSYN.PWY_final = subset(nonzero_COBALSYN.PWY, hem != "Not recorded")

#png("COBALSYN.PWY_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_COBALSYN.PWY = ggplot(nonzero_COBALSYN.PWY_final, aes(x = rel_abun, y= hem, fill = hem)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_rb) + xlim(-0.75, 0.5) + theme_classic(base_size =16) + labs(fill = "Anemia status", title = "Adenosylcobalamin salvage from cobinamide I") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to normal median)") + theme(legend.position = "NONE")
print(density_COBALSYN.PWY)
#dev.off()
```

\newpage

### Pathway breakdown  

COBALSYN-PWY: adenosylcobalamin salvage from cobinamide I

```{r composition_tax_cobalsynpwy, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("COBALSYN-PWY: adenosylcobalamin salvage from cobinamide I", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`COBALSYN-PWY: adenosylcobalamin salvage from cobinamide I` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("COBALSYN-PWY: adenosylcobalamin salvage from cobinamide I\\|g__.*\\.", "", names(bar))
names(bar) = gsub("COBALSYN-PWY: adenosylcobalamin salvage from cobinamide I\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$hem = stat_meta[ match(m$variable, rownames(stat_meta) ), "haemo_cat" ]
m = subset(m, hem != "Not recorded")

m$hem = factor(m$hem, levels = c("Normal", "Anemia"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_adenosylcobalamin, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~hem, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Adenosylcobalamin salvage from cobinamide I") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

2.1.1.130: Precorrin-2 C(20)-methyltransferase     

```{r precorrin2_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

precorrin2 = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.130: Precorrin-2 C(20)-methyltransferase`)
names(precorrin2) = c("Samples", "rel_abun")

nonzero_precorrin2 = subset(precorrin2, rel_abun != 0)
nonzero_precorrin2$diagnosis = stat_meta[match(nonzero_precorrin2$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_precorrin2$inflammation = stat_meta[match(nonzero_precorrin2$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_precorrin2, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_precorrin2$rel_abun = log10(nonzero_precorrin2$rel_abun/median_value)


nonzero_precorrin2$diagnosis = factor(nonzero_precorrin2$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_precorrin2_final = subset(nonzero_precorrin2, diagnosis != "NIJP" & diagnosis != "PsA")

#png("precorrin2_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_precorrin2 = ggplot(nonzero_precorrin2_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.25, 1) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Precorrin-2 C(20)-methyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_precorrin2)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.1.130: Precorrin-2 C(20)-methyltransferase

```{r composition_tax_211130, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.130: Precorrin-2", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.130: Precorrin-2 C(20)-methyltransferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.130: Precorrin-2 C\\(20)-methyltransferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.130: Precorrin-2 C\\(20)-methyltransferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_precorrin2, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Precorrin-2 C(20)-methyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

6.3.5.9: Hydrogenobyrinic acid a,c-diamide synthase (glutamine-hydrolyzing)  

```{r hydrogenobyrinic_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

hydrogenobyrinic = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`6.3.5.9: Hydrogenobyrinic acid a,c-diamide synthase (glutamine-hydrolyzing)`)
names(hydrogenobyrinic) = c("Samples", "rel_abun")

nonzero_hydrogenobyrinic = subset(hydrogenobyrinic, rel_abun != 0)
nonzero_hydrogenobyrinic$diagnosis = stat_meta[match(nonzero_hydrogenobyrinic$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_hydrogenobyrinic$inflammation = stat_meta[match(nonzero_hydrogenobyrinic$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_hydrogenobyrinic, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_hydrogenobyrinic$rel_abun = log10(nonzero_hydrogenobyrinic$rel_abun/median_value)


nonzero_hydrogenobyrinic$diagnosis = factor(nonzero_hydrogenobyrinic$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_hydrogenobyrinic_final = subset(nonzero_hydrogenobyrinic, diagnosis != "NIJP" & diagnosis != "PsA")

#png("hydrogenobyrinic_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_hydrogenobyrinic = ggplot(nonzero_hydrogenobyrinic_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.25, 1) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Hydrogenobyrinic acid a,c-diamide synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_hydrogenobyrinic)
#dev.off()
```

\newpage

### Pathway breakdown  

6.3.5.9: Hydrogenobyrinic acid a,c-diamide synthase (glutamine-hydrolyzing)

```{r composition_tax_6359, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("6.3.5.9: Hydrogenobyrinic acid a,c-diamide synthase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`6.3.5.9: Hydrogenobyrinic acid a,c-diamide synthase (glutamine-hydrolyzing)` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("6.3.5.9: Hydrogenobyrinic acid a,c-diamide synthase \\(glutamine-hydrolyzing)\\|g__.*\\.", "", names(bar))
names(bar) = gsub("6.3.5.9: Hydrogenobyrinic acid a,c-diamide synthase \\(glutamine-hydrolyzing)\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_hydrogenobyrinic, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Hydrogenobyrinic acid a,c-diamide synthase (glutamine-hydrolyzing)") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

2.7.1.156: Adenosylcobinamide kinase      

```{r adenosylcobinamide_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

adenosylcobinamide = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.7.1.156: Adenosylcobinamide kinase`)
names(adenosylcobinamide) = c("Samples", "rel_abun")

nonzero_adenosylcobinamide = subset(adenosylcobinamide, rel_abun != 0)
nonzero_adenosylcobinamide$diagnosis = stat_meta[match(nonzero_adenosylcobinamide$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_adenosylcobinamide$inflammation = stat_meta[match(nonzero_adenosylcobinamide$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_adenosylcobinamide, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_adenosylcobinamide$rel_abun = log10(nonzero_adenosylcobinamide$rel_abun/median_value)


nonzero_adenosylcobinamide$diagnosis = factor(nonzero_adenosylcobinamide$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_adenosylcobinamide_final = subset(nonzero_adenosylcobinamide, diagnosis != "NIJP" & diagnosis != "PsA")

#png("adenosylcobinamide_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_adenosylcobinamide = ggplot(nonzero_adenosylcobinamide_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.2, .75) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Adenosylcobinamide kinase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_adenosylcobinamide)
#dev.off()
```

\newpage

### Pathway breakdown  

2.7.1.156: Adenosylcobinamide kinase

```{r composition_tax_271156, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.7.1.156: Adenosylcobinamide kinase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.7.1.156: Adenosylcobinamide kinase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.7.1.156: Adenosylcobinamide kinase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.7.1.156: Adenosylcobinamide kinase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_adenosylcobinamide, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Adenosylcobinamide kinase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

2.1.1.132: Precorrin-6B C(5,15)-methyltransferase (decarboxylating)       

```{r precorrin6_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

precorrin6 = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.132: Precorrin-6B C(5,15)-methyltransferase (decarboxylating)`)
names(precorrin6) = c("Samples", "rel_abun")

nonzero_precorrin6 = subset(precorrin6, rel_abun != 0)
nonzero_precorrin6$diagnosis = stat_meta[match(nonzero_precorrin6$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_precorrin6$inflammation = stat_meta[match(nonzero_precorrin6$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_precorrin6, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_precorrin6$rel_abun = log10(nonzero_precorrin6$rel_abun/median_value)


nonzero_precorrin6$diagnosis = factor(nonzero_precorrin6$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_precorrin6_final = subset(nonzero_precorrin6, diagnosis != "NIJP" & diagnosis != "PsA")

#png("precorrin6_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_precorrin6 = ggplot(nonzero_precorrin6_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.25, 1) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Precorrin-6B C(5,15)-methyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_precorrin6)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.1.132: Precorrin-6B C(5,15)-methyltransferase (decarboxylating)

```{r composition_tax_211132, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.132: Precorrin-6B", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.132: Precorrin-6B C(5,15)-methyltransferase (decarboxylating)` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.132: Precorrin-6B C\\(5,15)-methyltransferase \\(decarboxylating)\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.132: Precorrin-6B C\\(5,15)-methyltransferase \\(decarboxylating)\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_percorrin6b, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Precorrin-6B C(5,15)-methyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

2.7.7.62: Adenosylcobinamide-phosphate guanylyltransferase     

```{r guanylyltransferase_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

guanylyltransferase = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.7.7.62: Adenosylcobinamide-phosphate guanylyltransferase`)
names(guanylyltransferase) = c("Samples", "rel_abun")

nonzero_guanylyltransferase = subset(guanylyltransferase, rel_abun != 0)
nonzero_guanylyltransferase$diagnosis = stat_meta[match(nonzero_guanylyltransferase$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_guanylyltransferase$inflammation = stat_meta[match(nonzero_guanylyltransferase$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_guanylyltransferase, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_guanylyltransferase$rel_abun = log10(nonzero_guanylyltransferase$rel_abun/median_value)


nonzero_guanylyltransferase$diagnosis = factor(nonzero_guanylyltransferase$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_guanylyltransferase_final = subset(nonzero_guanylyltransferase, diagnosis != "NIJP" & diagnosis != "PsA")

#png("guanylyltransferase_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_guanylyltransferase = ggplot(nonzero_guanylyltransferase_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "guanylyltransferase synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_guanylyltransferase)
#dev.off()
```

\newpage

### Pathway breakdown  

2.7.7.62: Adenosylcobinamide-phosphate guanylyltransferase

```{r composition_tax_27762, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.7.7.62: Adenosylcobinamide-phosphate guanylyltransferase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.7.7.62: Adenosylcobinamide-phosphate guanylyltransferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.7.7.62: Adenosylcobinamide-phosphate guanylyltransferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.7.7.62: Adenosylcobinamide-phosphate guanylyltransferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_guanylyltransferase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title=" Adenosylcobinamide-phosphate guanylyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

2.1.1.133: Precorrin-4 C(11)-methyltransferase

```{r precorrin4_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

precorrin4 = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.133: Precorrin-4 C(11)-methyltransferase`)
names(precorrin4) = c("Samples", "rel_abun")

nonzero_precorrin4 = subset(precorrin4, rel_abun != 0)
nonzero_precorrin4$diagnosis = stat_meta[match(nonzero_precorrin4$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_precorrin4$inflammation = stat_meta[match(nonzero_precorrin4$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_precorrin4, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_precorrin4$rel_abun = log10(nonzero_precorrin4$rel_abun/median_value)


nonzero_precorrin4$diagnosis = factor(nonzero_precorrin4$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_precorrin4_final = subset(nonzero_precorrin4, diagnosis != "NIJP" & diagnosis != "PsA")

#png("precorrin4_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_precorrin4 = ggplot(nonzero_precorrin4_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "precorrin4 synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_precorrin4)
#dev.off()
```


\newpage

### Pathway breakdown  

2.1.1.133: Precorrin-4 C(11)-methyltransferase

```{r composition_tax_211133, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.133: Precorrin-4", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.133: Precorrin-4 C(11)-methyltransferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.133: Precorrin-4 C\\(11)-methyltransferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.133: Precorrin-4 C\\(11)-methyltransferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_precorrin4, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Precorrin-4 C(11)-methyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

1.3.1.54: Precorrin-6A reductase     

```{r precorrin6a_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

precorrin6a = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`1.3.1.54: Precorrin-6A reductase`)
names(precorrin6a) = c("Samples", "rel_abun")

nonzero_precorrin6a = subset(precorrin6a, rel_abun != 0)
nonzero_precorrin6a$diagnosis = stat_meta[match(nonzero_precorrin6a$Samples, row.names(stat_meta)), "diagnosis"]
nonzero_precorrin6a$inflammation = stat_meta[match(nonzero_precorrin6a$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_precorrin6a, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_precorrin6a$rel_abun = log10(nonzero_precorrin6a$rel_abun/median_value)


nonzero_precorrin6a$diagnosis = factor(nonzero_precorrin6a$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_precorrin6a_final = subset(nonzero_precorrin6a, diagnosis != "NIJP" & diagnosis != "PsA")

#png("precorrin6a_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_precorrin6a = ggplot(nonzero_precorrin6a_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Precorrin-6A reductase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_precorrin6a)
#dev.off()
```

\newpage

### Pathway breakdown  

1.3.1.54: Precorrin-6A reductase

```{r composition_tax_13154, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("1.3.1.54: Precorrin-6A reductase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`1.3.1.54: Precorrin-6A reductase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("1.3.1.54: Precorrin-6A reductase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("1.3.1.54: Precorrin-6A reductase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "PsA" & diagnosis != "NIJP")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_precorrin6a, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Precorrin-6A reductase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

2.1.1.131: Precorrin-3B C(17)-methyltransferase    

```{r precorrin3b_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

precorrin3b = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.1.1.131: Precorrin-3B C(17)-methyltransferase`)
names(precorrin3b) = c("Samples", "rel_abun")

nonzero_precorrin3b = subset(precorrin3b, rel_abun != 0)
nonzero_precorrin3b$inflammation = stat_meta[match(nonzero_precorrin3b$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_precorrin3b, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_precorrin3b$rel_abun = log10(nonzero_precorrin3b$rel_abun/median_value)

nonzero_precorrin3b$inflammation = factor(nonzero_precorrin3b$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_precorrin3b_final = subset(nonzero_precorrin3b, inflammation != "Not recorded")

#png("precorrin3b_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_precorrin3b = ggplot(nonzero_precorrin3b_final, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Precorrin-3B C(17)-methyltransferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_precorrin3b)
#dev.off()
```

\newpage

### Pathway breakdown  

2.1.1.131: Precorrin-3B C(17)-methyltransferase

```{r composition_tax_211131, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.1.1.131: Precorrin-3B", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.1.1.131: Precorrin-3B C(17)-methyltransferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.1.1.131: Precorrin-3B C\\(17)-methyltransferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.1.1.131: Precorrin-3B C\\(17)-methyltransferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_precorrin3b, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Precorrin-3B C(17)-methyltransferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage  

3.1.3.73: Adenosylcobalamin/alpha-ribazole phosphatase                       

```{r ribazole_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

ribazole = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`3.1.3.73: Adenosylcobalamin/alpha-ribazole phosphatase`)
names(ribazole) = c("Samples", "rel_abun")

nonzero_ribazole = subset(ribazole, rel_abun != 0)
nonzero_ribazole$hem = stat_meta[match(nonzero_ribazole$Samples, row.names(stat_meta)), "haemo_cat"]
tmp = subset(nonzero_ribazole, hem == "Normal")
median_value = median(tmp$rel_abun)
nonzero_ribazole$rel_abun = log10(nonzero_ribazole$rel_abun/median_value)


nonzero_ribazole$hem = factor(nonzero_ribazole$hem, levels = c("Normal", "Anemia", "Not recorded"))
nonzero_ribazole_final = subset(nonzero_ribazole, hem != "Not recorded")

#png("ribazole_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_ribazole = ggplot(nonzero_ribazole_final, aes(x = rel_abun, y= hem, fill = hem)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_rb) + xlim(-1, .5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Adenosylcobalamin/alpha-ribazole phosphatase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_ribazole)
#dev.off()
```

\newpage

### Pathway breakdown  

3.1.3.73: Adenosylcobalamin/alpha-ribazole phosphatase

```{r composition_tax_31373, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("3.1.3.73: Adenosylcobalamin/alpha-ribazole phosphatase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`3.1.3.73: Adenosylcobalamin/alpha-ribazole phosphatase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("3.1.3.73: Adenosylcobalamin/alpha-ribazole phosphatase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("3.1.3.73: Adenosylcobalamin/alpha-ribazole phosphatase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$hem = stat_meta[ match(m$variable, rownames(stat_meta) ), "haemo_cat" ]
m = subset(m, hem != "Not recorded")

m$hem = factor(m$hem, levels = c("Normal", "Anemia"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_ribazole, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~hem, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Adenosylcobalamin/alpha-ribazole phosphatase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

## Iron sequestration

\newpage 

ENTBACSYN.PWY..enterobactin.biosynthesis  

```{r ENTBACSYN.PWY_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

ENTBACSYN.PWY = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`ENTBACSYN-PWY: enterobactin biosynthesis`)
names(ENTBACSYN.PWY) = c("Samples", "rel_abun")

nonzero_ENTBACSYN.PWY = subset(ENTBACSYN.PWY, rel_abun != 0)

nonzero_ENTBACSYN.PWY$inflammation = stat_meta[match(nonzero_ENTBACSYN.PWY$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_ENTBACSYN.PWY, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_ENTBACSYN.PWY$rel_abun = log10(nonzero_ENTBACSYN.PWY$rel_abun/median_value)

nonzero_ENTBACSYN.PWY$inflammation = factor(nonzero_ENTBACSYN.PWY$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_ENTBACSYN.PWY = subset(nonzero_ENTBACSYN.PWY, inflammation != "Not recorded")

density_ENTBACSYN.PWY_inflamm = ggplot(nonzero_ENTBACSYN.PWY, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 2) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Enterobactin biosynthesis") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_ENTBACSYN.PWY_inflamm)

```

\newpage

### Pathway breakdown  

ENTBACSYN-PWY: enterobactin biosynthesis

```{r composition_tax_entbacsynpwy, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("ENTBACSYN-PWY: enterobactin biosynthesis", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`ENTBACSYN-PWY: enterobactin biosynthesis` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("ENTBACSYN-PWY: enterobactin biosynthesis\\|g__.*\\.", "", names(bar))
names(bar) = gsub("ENTBACSYN-PWY: enterobactin biosynthesis\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_enterobatin, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="ENTBACSYN-PWY: enterobactin biosynthesis") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

Superpathway of heme biosynthesis from uroporphyrinogen-III  

```{r PWY0_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

PWY0 = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY0-1415: superpathway of heme biosynthesis from uroporphyrinogen-III`)
names(PWY0) = c("Samples", "rel_abun")

nonzero_PWY0 = subset(PWY0, rel_abun != 0)

nonzero_PWY0$inflammation = stat_meta[match(nonzero_PWY0$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_PWY0, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_PWY0$rel_abun = log10(nonzero_PWY0$rel_abun/median_value)

nonzero_PWY0$inflammation = factor(nonzero_PWY0$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_PWY0 = subset(nonzero_PWY0, inflammation != "Not recorded")

density_PWY0_inflamm = ggplot(nonzero_PWY0, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 2) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Superpathway of heme biosynthesis from uroporphyrinogen") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_PWY0_inflamm)

```

\newpage

### Pathway breakdown  

PWY0-1415: superpathway of heme biosynthesis from uroporphyrinogen-III

```{r composition_tax_pwy1415, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY0-1415: superpathway of heme biosynthesis from uroporphyrinogen-III", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY0-1415: superpathway of heme biosynthesis from uroporphyrinogen-III` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY0-1415: superpathway of heme biosynthesis from uroporphyrinogen-III\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY0-1415: superpathway of heme biosynthesis from uroporphyrinogen-III\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_uroporphyrinogen, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Superpathway of heme biosynthesis from uroporphyrinogen-III") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

PWY-5920: superpathway of heme biosynthesis from glycine   

```{r PWY5920_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

PWY5920 = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-5920: superpathway of heme biosynthesis from glycine`)
names(PWY5920) = c("Samples", "rel_abun")

nonzero_PWY5920 = subset(PWY5920, rel_abun != 0)

nonzero_PWY5920$inflammation = stat_meta[match(nonzero_PWY5920$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_PWY5920, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_PWY5920$rel_abun = log10(nonzero_PWY5920$rel_abun/median_value)

nonzero_PWY5920$inflammation = factor(nonzero_PWY5920$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_PWY5920 = subset(nonzero_PWY5920, inflammation != "Not recorded")

density_PWY5920_inflamm = ggplot(nonzero_PWY5920, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 2) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Superpathway of heme biosynthesis from glycine") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_PWY5920_inflamm)

```

\newpage

### Pathway breakdown  

PWY-5920: superpathway of heme biosynthesis from glycine

```{r composition_tax_pwy5920, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-5920: superpathway of heme biosynthesis from glycine", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-5920: superpathway of heme biosynthesis from glycine` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-5920: superpathway of heme biosynthesis from glycine\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-5920: superpathway of heme biosynthesis from glycine\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_heme-glycine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Superpathway of heme biosynthesis from glycine") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

HEME-BIOSYNTHESIS-II: heme biosynthesis I   

```{r heme_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

heme = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`HEME-BIOSYNTHESIS-II: heme biosynthesis I (aerobic)`)
names(heme) = c("Samples", "rel_abun")

nonzero_heme = subset(heme, rel_abun != 0)

nonzero_heme$inflammation = stat_meta[match(nonzero_heme$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_heme, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_heme$rel_abun = log10(nonzero_heme$rel_abun/median_value)

nonzero_heme$inflammation = factor(nonzero_heme$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_heme = subset(nonzero_heme, inflammation != "Not recorded")

density_heme_inflamm = ggplot(nonzero_heme, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 1.25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Heme biosynthesis I") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_heme_inflamm)

```

\newpage

### Pathway breakdown  

HEME-BIOSYNTHESIS-II: heme biosynthesis I (aerobic)

```{r composition_tax_hemebioII, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("HEME-BIOSYNTHESIS-II: heme biosynthesis I", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`HEME-BIOSYNTHESIS-II: heme biosynthesis I (aerobic)` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("HEME-BIOSYNTHESIS-II: heme biosynthesis I \\(aerobic)\\|g__.*\\.", "", names(bar))
names(bar) = gsub("HEME-BIOSYNTHESIS-II: heme biosynthesis I \\(aerobic)\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_heme-bioII, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Heme biosynthesis I (aerobic)") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

1.16.3.2: Bacterial non-heme ferritin  

```{r nonheme_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

nonheme = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`1.16.3.2: Bacterial non-heme ferritin`)
names(nonheme) = c("Samples", "rel_abun")

nonzero_nonheme = subset(nonheme, rel_abun != 0)

nonzero_nonheme$inflammation = stat_meta[match(nonzero_nonheme$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_nonheme, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_nonheme$rel_abun = log10(nonzero_nonheme$rel_abun/median_value)

nonzero_nonheme$inflammation = factor(nonzero_nonheme$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_nonheme = subset(nonzero_nonheme, inflammation != "Not recorded")

density_nonheme_inflamm = ggplot(nonzero_nonheme, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1, 1) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Bacterial non-heme ferritin") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_nonheme_inflamm)

```

\newpage

### Pathway breakdown  

1.16.3.2: Bacterial non-heme ferritin

```{r composition_tax_11632, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("1.16.3.2: Bacterial non-heme ferritin", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`1.16.3.2: Bacterial non-heme ferritin` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("1.16.3.2: Bacterial non-heme ferritin\\|g__.*\\.", "", names(bar))
names(bar) = gsub("1.16.3.2: Bacterial non-heme ferritin\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_non-heme_ferritin, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Bacterial non-heme ferritin") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

3.1.3.25: Inositol-phosphate phosphatase     

```{r inositol_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

inositol = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`3.1.3.25: Inositol-phosphate phosphatase`)
names(inositol) = c("Samples", "rel_abun")

nonzero_inositol = subset(inositol, rel_abun != 0)

nonzero_inositol$inflammation = stat_meta[match(nonzero_inositol$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_inositol, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_inositol$rel_abun = log10(nonzero_inositol$rel_abun/median_value)

nonzero_inositol$inflammation = factor(nonzero_inositol$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_inositol = subset(nonzero_inositol, inflammation != "Not recorded")

density_inositol_inflamm = ggplot(nonzero_inositol, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1, 1.25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Inositol-phosphate phosphatase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_inositol_inflamm)

```

\newpage

### Pathway breakdown  

3.1.3.25: Inositol-phosphate phosphatase

```{r composition_tax_31325, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("3.1.3.25: Inositol-phosphate phosphatase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`3.1.3.25: Inositol-phosphate phosphatase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("3.1.3.25: Inositol-phosphate phosphatase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("3.1.3.25: Inositol-phosphate phosphatase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_inositol, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Inositol-phosphate phosphatase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

1.15.1.2: Superoxide reductase    

```{r superoxide_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

superoxide = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`1.15.1.2: Superoxide reductase`)
names(superoxide) = c("Samples", "rel_abun")

nonzero_superoxide = subset(superoxide, rel_abun != 0)

nonzero_superoxide$inflammation = stat_meta[match(nonzero_superoxide$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_superoxide, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_superoxide$rel_abun = log10(nonzero_superoxide$rel_abun/median_value)

nonzero_superoxide$inflammation = factor(nonzero_superoxide$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_superoxide = subset(nonzero_superoxide, inflammation != "Not recorded")

density_superoxide_inflamm = ggplot(nonzero_superoxide, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 1) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Superoxide reductase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_superoxide_inflamm)

```

\newpage

### Pathway breakdown  

1.15.1.2: Superoxide reductase

```{r composition_tax_11512, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("1.15.1.2: Superoxide reductase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`1.15.1.2: Superoxide reductase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("1.15.1.2: Superoxide reductase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("1.15.1.2: Superoxide reductase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_superoxide, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Superoxide reductase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

3.6.3.41: Heme-transporting ATPase    

```{r heme_trans_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

heme_trans = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`3.6.3.41: Heme-transporting ATPase`)
names(heme_trans) = c("Samples", "rel_abun")

nonzero_heme_trans = subset(heme_trans, rel_abun != 0)

nonzero_heme_trans$inflammation = stat_meta[match(nonzero_heme_trans$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_heme_trans, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_heme_trans$rel_abun = log10(nonzero_heme_trans$rel_abun/median_value)

nonzero_heme_trans$inflammation = factor(nonzero_heme_trans$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_heme_trans = subset(nonzero_heme_trans, inflammation != "Not recorded")

density_heme_trans_inflamm = ggplot(nonzero_heme_trans, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.25, 1.25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Heme-transporting ATPase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_heme_trans_inflamm)

```

\newpage

### Pathway breakdown  

3.6.3.41: Heme-transporting ATPase

```{r composition_tax_36341, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("3.6.3.41: Heme-transporting ATPase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`3.6.3.41: Heme-transporting ATPase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("3.6.3.41: Heme-transporting ATPase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("3.6.3.41: Heme-transporting ATPase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_ATPase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Heme-transporting ATPase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

4.99.1.1: Ferrochelatase    

```{r ferrochelatase_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

ferrochelatase = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`4.99.1.1: Ferrochelatase`)
names(ferrochelatase) = c("Samples", "rel_abun")

nonzero_ferrochelatase = subset(ferrochelatase, rel_abun != 0)

nonzero_ferrochelatase$inflammation = stat_meta[match(nonzero_ferrochelatase$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_ferrochelatase, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_ferrochelatase$rel_abun = log10(nonzero_ferrochelatase$rel_abun/median_value)

nonzero_ferrochelatase$inflammation = factor(nonzero_ferrochelatase$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_ferrochelatase = subset(nonzero_ferrochelatase, inflammation != "Not recorded")

density_ferrochelatase_inflamm = ggplot(nonzero_ferrochelatase, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1, 1.5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Ferrochelatase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_ferrochelatase_inflamm)

```

\newpage

### Pathway breakdown  

4.99.1.1: Ferrochelatase

```{r composition_tax_49911, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("4.99.1.1: Ferrochelatase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`4.99.1.1: Ferrochelatase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("4.99.1.1: Ferrochelatase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("4.99.1.1: Ferrochelatase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")
m$Taxa = gsub("s__", "", m$Taxa)
m$Taxa = gsub("_1_1_47", "", m$Taxa)
m$Taxa = gsub("_", " ", m$Taxa)


m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

who = gsub("s__", "", who)
who = gsub("_1_1_47", "",who)
who = gsub("_", " ", who)
who = factor(who, levels = c("Escherichia coli", "Sutterella wadsworthensis", "Haemophilus parainfluenzae", "Burkholderiales bacterium", "Streptococcus salivarius", "Streptococcus parasanguinis", "Streptococcus thermophilus", "Actinomyces odontolyticus", "Eggerthella lenta", "unclassified", "Other"))
m$Taxa = factor(m$Taxa, levels = c("Escherichia coli", "Sutterella wadsworthensis", "Haemophilus parainfluenzae", "Burkholderiales bacterium", "Streptococcus salivarius", "Streptococcus parasanguinis", "Streptococcus thermophilus", "Actinomyces odontolyticus", "Eggerthella lenta", "unclassified", "Other"))
col_taxa = c("Escherichia coli" = "#2E8B57", "Sutterella wadsworthensis" = "#556B2F", "Haemophilus parainfluenzae" = "#3CB371", "Burkholderiales bacterium" = "#8FBC8F", "Streptococcus salivarius" = "#800080", "Streptococcus parasanguinis" = "#9370DB", "Streptococcus thermophilus" = "#BA55D3", "Actinomyces odontolyticus" = "#4682B4", "Eggerthella lenta" = "#1E90FF", "unclassified" = "#DCDCDC", "Other" = "#808080")

sub = subset(m, Taxa == "Escherichia coli")
sub = sub %>% arrange(inflammation, -value)
sub = sub$variable
sub = as.character(sub)

sub2 = subset(m, Taxa == "Streptococcus salivarius")
sub2 = sub2 %>% arrange(inflammation, -value)
list = intersect(sub, sub2$variable)
sub2 = sub2[-which(sub2$variable %in% list), ]
sub2 = sub2$variable
sub2 = as.character(sub2)

order_bar = append(sub, sub2)

sub3 = subset(m, Taxa == "Sutterella wadsworthensis")
sub3 = sub3 %>% arrange(inflammation, -value)
list = intersect(order_bar, sub3$variable)
sub3 = sub3[-which(sub3$variable %in% list), ]
sub3 = sub3$variable
sub3 = as.character(sub3)

order_bar = append(order_bar, sub3)
extras = setdiff(row.names(stat_meta), order_bar)

order_bar = append(order_bar, extras)

m$variable = factor(m$variable, levels = order_bar)
```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_ferrochelatase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, y = value, fill = Taxa)) + geom_bar(stat = "identity", position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  scale_fill_manual(values = col_taxa) + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Ferrochelatase") +
  theme(plot.title = element_text(size = 16))
plot(p)
#dev.off()
```

## Other functions
### Vitamin B12  

\newpage 

PWY-6895: superpathway of thiamin diphosphate biosynthesis II  

```{r thiamin_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

thiamin = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-6895: superpathway of thiamin diphosphate biosynthesis II`)
names(thiamin) = c("Samples", "rel_abun")

nonzero_thiamin = subset(thiamin, rel_abun != 0)
nonzero_thiamin$diagnosis = stat_meta[match(nonzero_thiamin$Samples, row.names(stat_meta)), "diagnosis"]
tmp = subset(nonzero_thiamin, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_thiamin$rel_abun = log10(nonzero_thiamin$rel_abun/median_value)


nonzero_thiamin$diagnosis = factor(nonzero_thiamin$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_thiamin_final = subset(nonzero_thiamin, diagnosis != "NIJP" & diagnosis != "PsA")

#png("thiamin_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_thiamin = ggplot(nonzero_thiamin_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.75, 2) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Thiamin diphosphate biosynthesis II") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_thiamin)
#dev.off()
```

***Only encoded by unclassified bugs***

\newpage

PWY-7204: pyridoxal 5'-phosphate salvage II (plants)  

```{r pyridoxal_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

pyridoxal = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-7204: pyridoxal 5'-phosphate salvage II (plants)`)
names(pyridoxal) = c("Samples", "rel_abun")

nonzero_pyridoxal = subset(pyridoxal, rel_abun != 0)
nonzero_pyridoxal$diagnosis = stat_meta[match(nonzero_pyridoxal$Samples, row.names(stat_meta)), "diagnosis"]
tmp = subset(nonzero_pyridoxal, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_pyridoxal$rel_abun = log10(nonzero_pyridoxal$rel_abun/median_value)


nonzero_pyridoxal$diagnosis = factor(nonzero_pyridoxal$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_pyridoxal_final = subset(nonzero_pyridoxal, diagnosis != "NIJP" & diagnosis != "PsA")

#png("pyridoxal_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_pyridoxal = ggplot(nonzero_pyridoxal_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-1.75, 2.2) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Pyridoxal 5'-phosphate salvage II") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_pyridoxal)
#dev.off()
```

***s__Klebsiella_sp_MS_92_3 or unclass***

\newpage

2.8.1.6: Biotin synthase

```{r biotin_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

biotin = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.8.1.6: Biotin synthase`)
names(biotin) = c("Samples", "rel_abun")

nonzero_biotin = subset(biotin, rel_abun != 0)
nonzero_biotin$diagnosis = stat_meta[match(nonzero_biotin$Samples, row.names(stat_meta)), "diagnosis"]
tmp = subset(nonzero_biotin, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_biotin$rel_abun = log10(nonzero_biotin$rel_abun/median_value)


nonzero_biotin$diagnosis = factor(nonzero_biotin$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_biotin_final = subset(nonzero_biotin, diagnosis != "NIJP" & diagnosis != "PsA")

#png("biotin_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_biotin = ggplot(nonzero_biotin_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.75, 0.75) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Biotin synthase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_biotin)
#dev.off()
```


```{r composition_tax_biotin, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.8.1.6: Biotin synthase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.8.1.6: Biotin synthase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.8.1.6: Biotin synthase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.8.1.6: Biotin synthase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_biotin, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Biotin synthase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

### L-tryosine
(L-methionine + Arginine )

Superpathway of L-tyrosine biosynthesis

```{r tyrosine_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

tyrosine = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-6630: superpathway of L-tyrosine biosynthesis`)
names(tyrosine) = c("Samples", "rel_abun")

nonzero_tyrosine = subset(tyrosine, rel_abun != 0)

nonzero_tyrosine$inflammation = stat_meta[match(nonzero_tyrosine$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_tyrosine, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_tyrosine$rel_abun = log10(nonzero_tyrosine$rel_abun/median_value)

nonzero_tyrosine$inflammation = factor(nonzero_tyrosine$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_tyrosine = subset(nonzero_tyrosine, inflammation != "Not recorded")

density_tyrosine_inflamm = ggplot(nonzero_tyrosine, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 2) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Superpathway of L-tyrosine biosynthesis") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_tyrosine_inflamm)

```

```{r composition_tax_tyrosine, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-6630: superpathway of L-tyrosine biosynthesis", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-6630: superpathway of L-tyrosine biosynthesis` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-6630: superpathway of L-tyrosine biosynthesis\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-6630: superpathway of L-tyrosine biosynthesis\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_tyrosine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="L-tyrosine biosynthesis") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```




\newpage 

### L-arginine

ARGSYNBSUB-PWY: L-arginine biosynthesis II (acetyl cycle)

```{r arginine_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

arginine = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`ARGSYNBSUB-PWY: L-arginine biosynthesis II (acetyl cycle)`)
names(arginine) = c("Samples", "rel_abun")

nonzero_arginine = subset(arginine, rel_abun != 0)

nonzero_arginine$inflammation = stat_meta[match(nonzero_arginine$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_arginine, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_arginine$rel_abun = log10(nonzero_arginine$rel_abun/median_value)

nonzero_arginine$inflammation = factor(nonzero_arginine$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_arginine = subset(nonzero_arginine, inflammation != "Not recorded")

density_arginine_inflamm = ggplot(nonzero_arginine, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1, 1) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "L-arginine biosynthesis II (acetyl cycle)") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_arginine_inflamm)

```

```{r composition_tax_arginine, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("ARGSYNBSUB-PWY: L-arginine biosynthesis II", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`ARGSYNBSUB-PWY: L-arginine biosynthesis II (acetyl cycle)` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("ARGSYNBSUB-PWY: L-arginine biosynthesis II \\(acetyl cycle)\\|g__.*\\.", "", names(bar))
names(bar) = gsub("ARGSYNBSUB-PWY: L-arginine biosynthesis II \\(acetyl cycle)\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_arginine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="L-arginine biosynthesis II (acetyl cycle)") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

### Purine degradation

PWY0-1297: superpathway of purine deoxyribonucleosides degradation

```{r purine_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY0-1297: superpathway of purine deoxyribonucleosides degradation`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1, 1.5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Purine deoxyribonucleosides degradation") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_purine, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY0-1297: superpathway of purine deoxyribonucleosides degradation", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY0-1297: superpathway of purine deoxyribonucleosides degradation` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY0-1297: superpathway of purine deoxyribonucleosides degradation\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY0-1297: superpathway of purine deoxyribonucleosides degradation\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_purine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Purine deoxyribonucleosides degradation") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

### NAD salvage pathway

NAD-BIOSYNTHESIS-II: NAD salvage pathway II

```{r NAD_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`NAD-BIOSYNTHESIS-II: NAD salvage pathway II`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.75, 2) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "NAD salvage pathway II") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_NAD, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("NAD-BIOSYNTHESIS-II: NAD salvage pathway II", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`NAD-BIOSYNTHESIS-II: NAD salvage pathway II` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("NAD-BIOSYNTHESIS-II: NAD salvage pathway II\\|g__.*\\.", "", names(bar))
names(bar) = gsub("NAD-BIOSYNTHESIS-II: NAD salvage pathway II\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_NAD, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="NAD salvage pathway II`") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

### Molybdenum cofactor biosynthesis

PWY-6823: molybdenum cofactor biosynthesis

```{r molybdenum_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-6823: molybdenum cofactor biosynthesis`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 3) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Molybdenum cofactor biosynthesis") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_molybdenum, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-6823: molybdenum cofactor biosynthesis", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-6823: molybdenum cofactor biosynthesis` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-6823: molybdenum cofactor biosynthesis\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-6823: molybdenum cofactor biosynthesis\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_molybdenum, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Molybdenum cofactor biosynthesis") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```



\newpage 

### Undecaprenyl

3.6.1.27: Undecaprenyl-diphosphate phosphatase

```{r undecaprenyl_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`3.6.1.27: Undecaprenyl-diphosphate phosphatase`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)
nonzero_density$diagnosis = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "diagnosis"]
tmp = subset(nonzero_density, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)


nonzero_density$diagnosis = factor(nonzero_density$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_density_final = subset(nonzero_density, diagnosis != "NIJP" & diagnosis != "PsA")

#png("density_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_density = ggplot(nonzero_density_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.2, 0.2) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Undecaprenyl-diphosphate phosphatase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_density)

```

```{r composition_tax_undecaprenyl, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("3.6.1.27: Undecaprenyl-diphosphate phosphatase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`3.6.1.27: Undecaprenyl-diphosphate phosphatase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("3.6.1.27: Undecaprenyl-diphosphate phosphatase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("3.6.1.27: Undecaprenyl-diphosphate phosphatase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "NIJP" & diagnosis != "PsA")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_undecaprenyl, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Molybdenum cofactor biosynthesis") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```



\newpage 

### Vitamin K

WY-5838: superpathway of menaquinol-8 biosynthesis I

```{r menaquinol_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-5838: superpathway of menaquinol-8 biosynthesis I`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 2) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Menaquinol-8 biosynthesis I") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_menaquinol, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-5838: superpathway of menaquinol-8 biosynthesis I", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-5838: superpathway of menaquinol-8 biosynthesis I` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-5838: superpathway of menaquinol-8 biosynthesis I\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-5838: superpathway of menaquinol-8 biosynthesis I\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_menaquinol, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Menaquinol-8 biosynthesis I") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```



\newpage 

### Treponoid

5.3.3.2: Isopentenyl-diphosphate Delta-isomerase

```{r isopentenyl_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`5.3.3.2: Isopentenyl-diphosphate Delta-isomerase`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 2) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Isopentenyl-diphosphate Delta-isomerase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_isopentenyl, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("5.3.3.2: Isopentenyl-diphosphate Delta-isomerase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`5.3.3.2: Isopentenyl-diphosphate Delta-isomerase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("5.3.3.2: Isopentenyl-diphosphate Delta-isomerase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("5.3.3.2: Isopentenyl-diphosphate Delta-isomerase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_isopentenyl, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Isopentenyl-diphosphate Delta-isomerase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

### Isoprenoid

PWY-7560: methylerythritol phosphate pathway II

```{r methylerythritol_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-7560: methylerythritol phosphate pathway II`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 1.5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Methylerythritol phosphate pathway II") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_methylerythritol, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-7560: methylerythritol phosphate pathway II", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-7560: methylerythritol phosphate pathway II` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-7560: methylerythritol phosphate pathway II\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-7560: methylerythritol phosphate pathway II\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_methylerythritol, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Methylerythritol phosphate pathway II") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage

PWY-6270: isoprene biosynthesis I   

```{r isoprene_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-6270: isoprene biosynthesis I`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 1.5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Isoprene biosynthesis I") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_isoprene, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-6270: isoprene biosynthesis I", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-6270: isoprene biosynthesis I` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-6270: isoprene biosynthesis I\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-6270: isoprene biosynthesis I\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_isoprene, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Isoprene biosynthesis I") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage

PWY-922: mevalonate pathway I 

```{r mevalonate_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-922: mevalonate pathway I`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-1.5, 2.25) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Mevalonate pathway I") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_mevalonate, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-922: mevalonate pathway I", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-922: mevalonate pathway I` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-922: mevalonate pathway I\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-922: mevalonate pathway I\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_mevalonate, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Mevalonate pathway I") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage

### Environmental response   

2.5.1.18: Glutathione transferase

```{r glutathione_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`2.5.1.18: Glutathione transferase`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)
nonzero_density$diagnosis = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "diagnosis"]
tmp = subset(nonzero_density, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)


nonzero_density$diagnosis = factor(nonzero_density$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_density_final = subset(nonzero_density, diagnosis != "NIJP" & diagnosis != "PsA")

#png("density_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_density = ggplot(nonzero_density_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-2, 3.5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Glutathione transferase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_density)

```

```{r composition_tax_glutathione, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("2.5.1.18: Glutathione transferase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`2.5.1.18: Glutathione transferase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("2.5.1.18: Glutathione transferase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("2.5.1.18: Glutathione transferase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "NIJP" & diagnosis != "PsA")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_glutathione, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Glutathione transferase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

3.5.2.6: Beta-lactamase

```{r betalactamase_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`3.5.2.6: Beta-lactamase`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)
nonzero_density$diagnosis = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "diagnosis"]
tmp = subset(nonzero_density, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)


nonzero_density$diagnosis = factor(nonzero_density$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_density_final = subset(nonzero_density, diagnosis != "NIJP" & diagnosis != "PsA")

#png("density_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_density = ggplot(nonzero_density_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.75, 0.75) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "Beta-lactamase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_density)

```

```{r composition_tax_betalactamase, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("3.5.2.6: Beta-lactamase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`3.5.2.6: Beta-lactamase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("3.5.2.6: Beta-lactamase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("3.5.2.6: Beta-lactamase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "NIJP" & diagnosis != "PsA")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_betalactamase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Beta-lactamase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```

\newpage 

5.99.1.2: DNA topoisomerase

```{r topoisomerase_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`5.99.1.3: DNA topoisomerase (ATP-hydrolyzing)`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)
nonzero_density$diagnosis = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "diagnosis"]
tmp = subset(nonzero_density, diagnosis == "HC")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)


nonzero_density$diagnosis = factor(nonzero_density$diagnosis, levels = c("HC", "RA", "AS", "PsA", "NIJP"))
nonzero_density_final = subset(nonzero_density, diagnosis != "NIJP" & diagnosis != "PsA")

#png("density_denisty_diagnosis.png", height = 3, width = 7, units = "in", res = 600)
density_density = ggplot(nonzero_density_final, aes(x = rel_abun, y= diagnosis, fill = diagnosis)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_diagnosis) + xlim(-0.5, 0.5) + theme_classic(base_size =16) + labs(fill = "Patient Diagnosis", title = "DNA topoisomerase (ATP-hydrolyzing)") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + xlab("log10(abundance relative to HC median)") + theme(legend.position = "NONE")
print(density_density)

```

```{r composition_tax_topoisomerase, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("5.99.1.3: DNA topoisomerase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`5.99.1.3: DNA topoisomerase (ATP-hydrolyzing)` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("5.99.1.3: DNA topoisomerase \\(ATP-hydrolyzing)\\|g__.*\\.", "", names(bar))
names(bar) = gsub("5.99.1.3: DNA topoisomerase \\(ATP-hydrolyzing)\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$diagnosis = dat_meta[ match(m$variable, rownames(dat_meta) ), "diagnosis" ]
m = subset(m, diagnosis != "NIJP" & diagnosis != "PsA")

m$diagnosis = factor(m$diagnosis, levels = c("HC", "RA", "AS"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_topoisomerase, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~diagnosis, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="DNA topoisomerase (ATP-hydrolyzing)") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

4.2.2.n1: Peptidoglycan lytic exotransglycosylase

```{r peptidoglycan_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_ecs_common), dat_dna_ecs_common$`4.2.2.n1: Peptidoglycan lytic exotransglycosylase`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 3) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Peptidoglycan lytic exotransglycosylase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_peptidoglycan, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("4.2.2.n1: Peptidoglycan lytic exotransglycosylase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`4.2.2.n1: Peptidoglycan lytic exotransglycosylase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("4.2.2.n1: Peptidoglycan lytic exotransglycosylase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("4.2.2.n1: Peptidoglycan lytic exotransglycosylase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_peptidoglycan, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Peptidoglycan lytic exotransglycosylase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

PWY-6803: phosphatidylcholine acyl editing

```{r phosphatidylcholine_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-6803: phosphatidylcholine acyl editing`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 2.75) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Phosphatidylcholine acyl editing") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_phosphatidylcholine, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-6803: phosphatidylcholine acyl editing", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-6803: phosphatidylcholine acyl editing` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-6803: phosphatidylcholine acyl editing\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-6803: phosphatidylcholine acyl editing\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_phosphatidylcholine, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Phosphatidylcholine acyl editing") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

PWY-561: superpathway of glyoxylate cycle and fatty acid degradation

```{r glyoxylate_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-561: superpathway of glyoxylate cycle and fatty acid degradation`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 2.75) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Glyoxylate cycle and fatty acid degradation") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_glyoxylate, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-561: superpathway of glyoxylate cycle and fatty acid degradation", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-561: superpathway of glyoxylate cycle and fatty acid degradation` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-561: superpathway of glyoxylate cycle and fatty acid degradation\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-561: superpathway of glyoxylate cycle and fatty acid degradation\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_glyoxylate, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Glyoxylate cycle and fatty acid degradation") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

PWY-7409: phospholipid remodeling (phosphatidylethanolamine, yeast)

```{r phospholipid_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-7409: phospholipid remodeling (phosphatidylethanolamine, yeast)`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 2.75) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Phospholipid remodeling") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_phospholipid, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-7409: phospholipid remodeling", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-7409: phospholipid remodeling (phosphatidylethanolamine, yeast)` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-7409: phospholipid remodeling \\(phosphatidylethanolamine, yeast)\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-7409: phospholipid remodeling \\(phosphatidylethanolamine, yeast)\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_phospholipid, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Glyoxylate cycle and fatty acid degradation") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

PWY-5705: allantoin degradation to glyoxylate III

```{r Allantoin_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-5705: allantoin degradation to glyoxylate III`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 2.75) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Allantoin degradation to glyoxylate III") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_allantoin, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-5705: allantoin degradation to glyoxylate III", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-5705: allantoin degradation to glyoxylate III` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-5705: allantoin degradation to glyoxylate III\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-5705: allantoin degradation to glyoxylate III\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_allantoin, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Allantoin degradation to glyoxylate III") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

PWY-6690: cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate

```{r cinnamate_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_path_common$`PWY-6690: cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2, 2.75) + theme_classic(base_size = 12)+ labs(fill = "Inflammation Status (CRP)", title = "Cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_cinnamate, include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_path[grep("PWY-6690: cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate", row.names(dat_dna_path), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`PWY-6690: cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("PWY-6690: cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate\\|g__.*\\.", "", names(bar))
names(bar) = gsub("PWY-6690: cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_cinnamate, echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Cinnamate and 3-hydroxycinnamate degradation to 2-oxopent-4-enoate") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```


\newpage 

5.5.1.1: Muconate cycloisomerase

```{r muconate_density_diagnosis, echo = FALSE, warning = FALSE, message = FALSE}

density = data.frame(row.names(dat_dna_path_common), dat_dna_ecs_common$`5.5.1.1: Muconate cycloisomerase`)
names(density) = c("Samples", "rel_abun")

nonzero_density = subset(density, rel_abun != 0)

nonzero_density$inflammation = stat_meta[match(nonzero_density$Samples, row.names(stat_meta)), "inflammation"]
tmp = subset(nonzero_density, inflammation == "Not inflamed")
median_value = median(tmp$rel_abun)
nonzero_density$rel_abun = log10(nonzero_density$rel_abun/median_value)

nonzero_density$inflammation = factor(nonzero_density$inflammation, levels = c("Not inflamed", "Inflammation", "Some inflammation",  "Not recorded"))
nonzero_density = subset(nonzero_density, inflammation != "Not recorded")

density_density_inflamm = ggplot(nonzero_density, aes(x = rel_abun, y= inflammation, fill = inflammation)) + geom_density_ridges(alpha = 0.6, scale = 15) + scale_fill_manual(values = col_inflammation) + xlim(-2.5, 0.5) + theme_classic(base_size = 16)+ labs(fill = "Inflammation Status (CRP)", title = "Muconate cycloisomerase") + ylab("Density")  + theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) + theme(legend.position = "NONE") + xlab("log10(abundance relative to HC median)")
print(density_density_inflamm)

```

```{r composition_tax_muconate , include=FALSE}
#Creating bar plots for visualizations
bar = dat_dna_ecs[grep("5.5.1.1: Muconate cycloisomerase", row.names(dat_dna_ecs), value = T), ]
bar = data.frame(t(bar), check.names = F)
bar$`5.5.1.1: Muconate cycloisomerase` = NULL
bar = bar/rowSums(bar)
bar = bar[complete.cases(bar), ]
names(bar) = gsub("5.5.1.1: Muconate cycloisomerase\\|g__.*\\.", "", names(bar))
names(bar) = gsub("5.5.1.1: Muconate cycloisomerase\\|", "", names(bar))
who = names(sort(colMeans(bar), decreasing = TRUE))[1:10]
f = bar[,names(bar) %in% who]
f = f[order(colMeans(f), decreasing = T)]
f$Other = 1-rowSums(f)
who = c(who, "Other")

f = f[order(f[1], decreasing = T ), ]
f = data.frame(t(f))
head(f)
f$Taxa = row.names(f)

m = melt(f)
m = subset(m, m$value > 0)

m$inflammation = dat_meta[ match(m$variable, rownames(dat_meta) ), "inflammation" ]
m = subset(m, inflammation != "Not recorded")

m$inflammation = factor(m$inflammation, levels = c("Not inflamed", "Some inflammation", "Inflammation"))

```

Barplot faceted by diagnosis  

```{r bar_species_by_diagnosis_muconate , echo = FALSE, warning = FALSE, message = FALSE, fig.height=4, fig.width= 13}
#png("p_copri_bar_graph.png", width = 15, height = 6, units = "in", res = 600)
p = ggplot(m, aes(variable, fill = Taxa)) + geom_bar(aes(weight = value), position = position_fill(reverse = TRUE)) + 
  theme_classic() + facet_grid(~inflammation, scales = "free", space ="free") + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + 
  col + xlab("Sample") + ylab("Relative Abundance") + theme(legend.text = element_text(face = "italic")) + guides(fill = guide_legend(ncol = 1, reverse=TRUE, keyheight = 0.55)) + scale_y_continuous(labels = percent_format()) + 
  labs(title="Muconate cycloisomerase") +
  theme(plot.title = element_text(size = 16))
p$data$Taxa = factor(p$data$Taxa, ordered = TRUE, levels = who)
plot(p)
#dev.off()
```